---
title: "Chilean soil communities"
author: "David Castro"
date: "`r Sys.Date()`"
output: 
  html_document:
    toc: true
    toc_float: true
    number_sections: true
---

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir=getwd())
```

# Libraries
```{r, echo =TRUE, message= FALSE}
library(ggplot2) #plots
library(dplyr) #data manipulation
library(plyr) #data manipulation
library(FSA) #Dunn's Test
library(rcompanion) #letter grouping for Dunn's test
library(phyloseq) #Metagenomic analysis
library(genefilter) #filter function
library(VennDiagram) #venn diagram
library(picante) #richness
library(pairwiseAdonis) #pairwise adonis analysis
library(clusterSim) #correlation plot
library(DESeq2) #differentially abundant taxa
library(gplots) #heatmap
library(pheatmap) #heatmaps
library(ComplexHeatmap) #heatmaps
library(circlize) #colors for Heatmap function
library(tidyr) #drop_na 
library(rwantshue) #color generator
library(RColorBrewer) #color palette
library(ggordiplots) #ordiplot
library(gridExtra) #grid arrange
```


#Fungal metagenomics
## Amplicon sequencing analysis

The amplicon sequencing analysis was made using q2 for pre-processing and phyloseq package for the phylogenetic analysis.

phyloseq object preparation
```{r}
# ASV table
ASVs <- read.table("ITS2/table_ITS2.txt", sep="\t", header=TRUE, row.names=1)
# taxonomy
taxonomy <- read.table("ITS2/taxonomy.tsv", sep="\t", header=TRUE, row.names=1, na.strings=c(""," ","NA"))
taxonomy <- as.matrix(taxonomy)
# tree
tree <- read_tree("ITS2/tree.nwk")
# sample data
metadata <- read.table("ITS2/metadata.tsv", sep="\t", header=TRUE, row.names=1, na.strings=c(""," ","NA"),dec = ",")
# phyloseq object
phy_obj <- phyloseq(otu_table(ASVs, taxa_are_rows = TRUE), 
                    tax_table(taxonomy), sample_data(metadata),tree)

tax_table(phy_obj) <- gsub(".__","",tax_table(phy_obj)) #separate taxonomy in columns

phy_obj_Fungi <- phy_obj
```

Filtering ASV equal or lower than 10. Filtering was made with filterfun() from the genefilter package to set the threshold and filter_taxa from phyloseq to remove the ASV lower than the set filter #revise
```{r}
flist    <- filterfun(kOverA(1, 10)) #remove AVS's equal or lower than 10 
phy_obj_Fungi <- filter_taxa(phy_obj_Fungi, flist, TRUE)
phy_obj_Fungi
sample_data(phy_obj_Fungi)
```

Filtering ASV with a minimum abundance of 0.005% per sample.
First the 0.005% had to be calculated for each sample type, by merging the phyloseq object by sample type (merge_samples()), we get the total number of reads (sample_sums()) and we can calculate the 0.005% for each. 
Then, the phyloseq object has to be splitted by sample type (subset_samples()) and the ASVs lower than the calculated 0.005% (abundance object) can be removed (prune_taxa()).
Finally, the phyloseq objects by sample type can be merged into one new phyloseq object (merge_phyloseq())
```{r}
Pch <- subset_samples(phy_obj_Fungi,Specie=="P_chilensis")

abundance <- as.data.frame(sample_sums(merge_samples(Pch,group = "Sample")))
colnames(abundance) <- "abundance"
abundance[,".005%"] <- abundance$abundance*.00005
abundance$`.005%` <- round(abundance$`.005%`)
abundance #minimum number of reads to remove per-sample
abundance <- t(abundance)[-1,]
abundance <- as.data.frame(t(abundance))
abundance

#Subseting by sample type
Root <- subset_samples(Pch,Sample=="Root")
Root
Soil <- subset_samples(Pch,Sample=="Soil")
Soil

#Removing ASV lower than 0.005% of abundance per sample type

Root <- prune_taxa(taxa_sums(Root) > abundance$Root,Root)
Root
Soil <- prune_taxa(taxa_sums(Soil) > abundance$Soil,Soil)
Soil

#preparing for merge into a new phy_obj 
Root <- phyloseq(otu_table(Root),sample_data(Root),tax_table(Root))
Soil <- phyloseq(otu_table(Soil),sample_data(Soil),tax_table(Soil))

Pch <- merge_phyloseq(Root,Soil)
Pch
```

```{r}
Pta <- subset_samples(phy_obj_Fungi,Specie=="P_tamarugo")

abundance <- as.data.frame(sample_sums(merge_samples(Pta,group = "Sample")))
colnames(abundance) <- "abundance"
abundance[,".005%"] <- abundance$abundance*.00005
abundance$`.005%` <- round(abundance$`.005%`)
abundance #minimum number of reads to remove per-sample
abundance <- t(abundance)[-1,]
abundance <- as.data.frame(t(abundance))
abundance

#Subseting by sample type
Root <- subset_samples(Pta,Sample=="Root")
Root
Soil <- subset_samples(Pta,Sample=="Soil")
Soil

#Removing ASV lower than 0.005% of abundance per sample type

Root <- prune_taxa(taxa_sums(Root) > abundance$Root,Root)
Root
Soil <- prune_taxa(taxa_sums(Soil) > abundance$Soil,Soil)
Soil

#preparing for merge into a new phy_obj 
Root <- phyloseq(otu_table(Root),sample_data(Root),tax_table(Root))
Soil <- phyloseq(otu_table(Soil),sample_data(Soil),tax_table(Soil))

Pta <- merge_phyloseq(Root,Soil)
Pta
```

```{r}
#Adding phy_tree
phy_obj_Fungi <- merge_phyloseq(Pch,Pta)
phy_obj_Fungi <- phyloseq(otu_table(phy_obj_Fungi),sample_data(phy_obj_Fungi),tax_table(phy_obj_Fungi),tree)
phy_obj_Fungi
```

Merging phylogenetic tree tips.
This is made to avoid many ASVs annotating for the same specie. Using the tip_glom() function, the tips of the phylogenetic tree lower than a given 'height' will be merged into one.
```{r}
phy_obj_Fungi <- tip_glom(phy_obj_Fungi,hcfun = hclust)
phy_obj_Fungi
```

Library size and normalization
```{r}
#Library size
sums <- sample_sums(phy_obj_Fungi)
barplot(sums[order(sums, decreasing=TRUE)], las=3, cex.names = 0.5, ylab="Library size", col = "skyblue")
abline(h = 10000, lty=2)

#Normalisation by proportions
phy_obj_Fungi <- prune_samples(sample_sums(phy_obj_Fungi)>10000,phy_obj_Fungi)
phy_obj_Fungi

phy_norm <- transform_sample_counts(phy_obj_Fungi,function(x) x/sum(x))
barplot(sample_sums(phy_norm),las=3,cex.names=.5,col="skyblue")

#Rarefy
set.seed(12345)
phy_raref <- rarefy_even_depth(phy_obj_Fungi, sample.size = min(sample_sums(phy_obj_Fungi)), replace = FALSE, trimOTUs = TRUE, verbose = TRUE)
barplot(sample_sums(phy_raref), las=3, cex.names = 0.5, col = "skyblue")
```

## Richness, alpha and beta diversity
### Richness 
Richness was calculated using the pd() function from picante package.
Preparing richness dataframe
```{r}
faiths_div <- pd(as.data.frame(t(otu_table(phy_raref))), phy_tree(phy_obj_Fungi), include.root = FALSE)
faiths_div <- merge(faiths_div,sample_data(phy_obj_Fungi), by = "row.names")
faiths_div$Soil <- factor(faiths_div$Soil, levels=c('Pozo_Almonte','La_Serena','Vicuna','Los_Angeles'))
faiths_div$Specie <- as.factor(faiths_div$Specie)
```

Plotting richness
```{r echo= FALSE}
ggplot(data=faiths_div %>% filter(Sample=="Soil"), aes(x = Sample, y = SR, fill = Soil)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("brown", "gold","darkgoldenrod3","darkorange")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "Specie richness")
```
```{r echo= FALSE}
ggplot(data=faiths_div %>% filter(Sample=="Root"), aes(x = Sample, y = SR, fill = Specie)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("gold","darkgoldenrod3","darkorange")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "Specie richness")
```

```{r echo= FALSE}
ggplot(faiths_div, aes(x = Specie, y = SR, fill = Soil)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("brown", "gold","darkgoldenrod3","darkorange")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "Specie richness",limits = c(0,80)) + 
  facet_grid(.~Sample)
```
Statistical analysis of richness using Kruskal-Wallis. Any significant differences were further analysed with Dunnâ€™s test with dunnTest() function from FAS package.
The model used was richness~Sample type, richness~ fertilization type and richness~sample*fertilization
```{r}
r_root <- faiths_div %>%
  filter(Sample == "Root")

agricolae::kruskal(r_root$SR, r_root$Specie, p.adj = "fdr", group = T, console = T)

agricolae::kruskal(r_root %>% filter(Specie=="P_chilensis") %>% dplyr::select(SR), r_root %>% filter(Specie=="P_chilensis") %>% dplyr::select(Soil), p.adj = "fdr", group = T, console = T)

agricolae::kruskal(r_root %>% filter(Specie=="P_tamarugo") %>% dplyr::select(SR), r_root %>% filter(Specie=="P_tamarugo") %>% dplyr::select(Soil), p.adj = "fdr", group = T, console = T)

r_soil <- faiths_div %>%
  filter(Sample == "Soil")

agricolae::kruskal(r_soil$SR, r_soil$Soil, p.adj = "fdr", group = T, console = T)
```

### Alpha diversity

The Alpha diversity was calculated using the Shannon diversity index with the function diversity() from the vegan package. 

Calculating Shannon diversity
```{r}
shannon_div <- vegan::diversity(otu_table(phy_obj_Fungi), MARGIN = 2, index = "shannon")
shannon_div <- as.data.frame(shannon_div)
shannon_div <- merge(shannon_div,sample_data(phy_obj_Fungi), by = "row.names")
shannon_div$Soil <- factor(shannon_div$Soil, levels=c('Pozo_Almonte','La_Serena','Vicuna','Los_Angeles'))
shannon_div$Specie <- as.factor(shannon_div$Specie)
```

Plotting alpha diversity index
```{r echo= FALSE}
ggplot(data=shannon_div %>% filter(Sample=="Soil"), aes(x = Treatment, y = shannon_div, fill = Soil)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("brown", "gold","darkgoldenrod3","darkorange")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "Shannon Diversity Index")
```

```{r echo= FALSE}
ggplot(shannon_div, aes(x = Specie, y = shannon_div, fill = Soil)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("brown", "gold","darkgoldenrod3","darkorange")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "Shannon Diversity Index",limits = c(0,3.5)) + 
  facet_grid(.~Sample)
```

Statistical analysis of Shannon diversity index using Kruskal-Wallis. Any significant differences were further analysed with Dunnâ€™s test with dunnTest() function from FAS package.
The model used was shannon~Sample type, shannon~fertilization type and shannon~sample*fertilization
```{r}
a_root <- shannon_div %>%
  filter(Sample == "Root")

agricolae::kruskal(a_root$shannon_div, a_root$Specie, p.adj = "fdr", group = T, console = T)

agricolae::kruskal(a_root %>% filter(Specie=="P_chilensis") %>% dplyr::select(shannon_div), a_root %>% filter(Specie=="P_chilensis") %>% dplyr::select(Soil), p.adj = "fdr", group = T, console = T)

agricolae::kruskal(a_root %>% filter(Specie=="P_tamarugo") %>% dplyr::select(shannon_div), a_root %>% filter(Specie=="P_tamarugo") %>% dplyr::select(Soil), p.adj = "fdr", group = T, console = T)

a_soil <- shannon_div %>%
  filter(Sample == "Soil")

agricolae::kruskal(a_soil$shannon_div, a_soil$Soil, p.adj = "fdr", group = T, console = T)
```

### Beta diversity
Beta diversity was evaluated using PCoA with bray-curtis distance

```{r echo= FALSE}
#Soil
ord <- ordinate(subset_samples(phy_raref, Sample == "Soil"), method = "PCoA", distance = "bray", weighted=FALSE)
sample_data(phy_norm)[,"Specie_sample"] <- paste(sample_data(phy_norm)$Specie,sample_data(phy_norm)$Sample, sep="-")

plot1<- gg_ordiplot(ord$vectors, groups = sample_data(subset_samples(phy_norm, Sample == "Soil"))$Soil, hull = TRUE, 
            label = TRUE, spiders = TRUE, ellipse = FALSE, pt.size = 7, plot = FALSE)

ggplot(data=plot1$df_spiders) + 
  geom_point(aes(x=x, y=y,shape=Group, colour=Group), size = 5) +
  geom_point(aes(x=cntr.x, y=cntr.y), size = 5, shape = 18) +
  scale_color_manual(values=c("gold", "darkorange", "brown", "darkgoldenrod3")) + scale_shape_manual(values = c(19,19,19,19)) +
  theme_bw() + 
  theme(legend.position = "bottom", legend.box = "vertical", 
        text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        panel.border = element_rect(size = 2, fill = NA))
```

```{r echo= FALSE}
phy_obj_F_Soil <- subset_samples(phy_obj_Fungi, Sample == "Soil")
phy_obj_F_Soil <- subset_samples(phy_obj_F_Soil, Soil != "Pozo_Almonte")

sample_data(phy_obj_F_Soil)[,"sample_site"] <- paste(sample_data(phy_obj_F_Soil)$Sample, sample_data(phy_obj_F_Soil)$Soil, sep = "_")

phy_norm_F_Soil <- subset_samples(phy_norm, Sample == "Soil")
phy_norm_F_Soil <- subset_samples(phy_norm_F_Soil, Soil != "Pozo_Almonte")

sample_data(phy_norm_F_Soil)[,"sample_site"] <- paste(sample_data(phy_norm_F_Soil)$Sample, sample_data(phy_norm_F_Soil)$Soil, sep = "_")

phy_raref_F_Soil <- subset_samples(phy_raref, Sample == "Soil")
phy_raref_F_Soil <- subset_samples(phy_raref_F_Soil, Soil != "Pozo_Almonte")

sample_data(phy_raref_F_Soil)[,"sample_site"] <- paste(sample_data(phy_raref_F_Soil)$Sample, sample_data(phy_raref_F_Soil)$Soil, sep = "_")

#Root
phy_obj_F_Root <- subset_samples(phy_obj_Fungi, Sample == "Root")
phy_norm_F_Root <- subset_samples(phy_norm, Sample == "Root")
phy_raref_F_Root <- subset_samples(phy_raref, Sample == "Root")

sample_data(phy_obj_F_Root)[,"sample_site"] <- paste(sample_data(phy_obj_F_Root)$Sample, sample_data(phy_obj_F_Root)$Soil, sample_data(phy_obj_F_Root)$Specie, sep = "_")

sample_data(phy_norm_F_Root)[,"sample_site"] <- paste(sample_data(phy_norm_F_Root)$Sample, sample_data(phy_norm_F_Root)$Soil, sample_data(phy_norm_F_Root)$Specie, sep = "_")

sample_data(phy_raref_F_Root)[,"sample_site"] <- paste(sample_data(phy_raref_F_Root)$Sample, sample_data(phy_raref_F_Root)$Soil, sample_data(phy_raref_F_Root)$Specie, sep = "_")

phy_obj2 <- merge_phyloseq(phy_obj_F_Soil,phy_obj_F_Root)

phy_norm2 <- merge_phyloseq(phy_norm_F_Soil,phy_norm_F_Root)

phy_raref2 <- merge_phyloseq(phy_raref_F_Soil,phy_raref_F_Root)

ord <- ordinate(phy_raref2, method = "PCoA", distance = "bray", weighted=FALSE)

plot_ordination(phy_norm2, ord, color = "Soil", shape = "sample_site") + geom_point(size=7) +
  scale_color_manual(values=c("gold", "darkorange", "darkgoldenrod3")) + scale_shape_manual(values = c(24,25,24,25,24,25,19,19,19,19)) +
  theme_bw() + 
  theme(legend.position = "bottom", legend.box = "vertical", 
        text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        panel.border = element_rect(size = 2, fill = NA))

plot1<- gg_ordiplot(ord$vectors, groups = sample_data(phy_norm2)$sample_site, hull = TRUE, 
            label = TRUE, spiders = TRUE, ellipse = FALSE, pt.size = 7, plot = FALSE)

#pdf("Manuscript/Plots/Fungal_beta_Both.pdf",height=10, width=10)
ggplot(data=plot1$df_spiders) + 
  geom_point(aes(x=x, y=y,shape=Group, colour=Group), size = 5) +
  geom_point(aes(x=cntr.x, y=cntr.y), size = 5, shape = 18) +
  scale_color_manual(values=c("gold","gold", "darkorange","darkorange", "darkgoldenrod3","darkgoldenrod3","gold", "darkorange", "darkgoldenrod3")) + scale_shape_manual(values = c(24,25,24,25,24,25,19,19,19,19)) +
  theme_bw() + 
  theme(legend.position = "bottom", legend.box = "vertical", 
        text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        panel.border = element_rect(size = 2, fill = NA))
#dev.off()

(sim <- with(sample_data(subset_samples(phy_obj2, Sample == "Root")), simper(t(otu_table(subset_samples(phy_obj2, Sample == "Root"))), sample_site)))
#summary(sim)
```

Statistical analysis of beta diversity using PerMANOVA with adonis() function from vegan package.
The model used was ASVs~Sample*fertilization
```{r}
tmp <- prune_samples(!is.na(sample_data(phy_obj2)$Specie), phy_obj2) 
adonis(t(otu_table(tmp)) ~ Specie * Sample * Soil, data = data.frame(sample_data(phy_obj2)))
#pairwise.adonis(t(otu_table(tmp)),data.frame(sample_data(phy_obj2))$Specie)
#pairwise.adonis(t(otu_table(tmp)),data.frame(sample_data(phy_obj2))$Sample)
#pairwise.adonis(t(otu_table(tmp)),data.frame(sample_data(phy_obj2))$Soil)
#pairwise.adonis(t(otu_table(tmp)),paste(data.frame(sample_data(phy_obj2))$Specie,data.frame(sample_data(phy_obj2))$Sample,sep = "."))
#pairwise.adonis(t(otu_table(tmp)),paste(data.frame(sample_data(phy_obj2))$Specie,data.frame(sample_data(phy_obj2))$Soil,sep = "."))
#pairwise.adonis(t(otu_table(tmp)),paste(data.frame(sample_data(phy_obj2))$Specie,data.frame(sample_data(phy_obj2))$Sample, data.frame(sample_data(phy_obj2))$Soil,sep = "."))
```

## Community composition
### Disimilarity Heatmap
``` {r echo= FALSE}
# Preparing data

taxonomy2 <- data.frame(paste(tax_table(subset_samples(phy_norm, Sample == "Soil"))[,1],tax_table(subset_samples(phy_norm, Sample == "Soil"))[,2],tax_table(subset_samples(phy_norm, Sample == "Soil"))[,3],tax_table(subset_samples(phy_norm, Sample == "Soil"))[,4],
                  tax_table(subset_samples(phy_norm, Sample == "Soil"))[,5],tax_table(subset_samples(phy_norm, Sample == "Soil"))[,6],tax_table(subset_samples(phy_norm, Sample == "Soil"))[,7],sep = ";"))
rownames(taxonomy2) <- rownames(tax_table(subset_samples(phy_norm, Sample == "Soil")))
colnames(taxonomy2) <- "taxonomy"
taxonomy2[,"ASV"] <- rownames(taxonomy2)

sim <- with(sample_data(subset_samples(phy_obj_Fungi, Sample == "Soil")), simper(t(otu_table(subset_samples(phy_obj_Fungi, Sample == "Soil"))), Soil))
#summary(sim)

df1 <- rownames(summary(sim)[[1]] %>% filter(cumsum < .7)) #LS_V
df2 <- rownames(summary(sim)[[2]] %>% filter(cumsum < .7)) #LS_PA
df3 <- rownames(summary(sim)[[3]] %>% filter(cumsum < .7)) #LS_LA
df4 <- rownames(summary(sim)[[4]] %>% filter(cumsum < .7)) #V_PA
df5 <- rownames(summary(sim)[[5]] %>% filter(cumsum < .7)) #V_LA
df6 <- rownames(summary(sim)[[6]] %>% filter(cumsum < .7)) #PA_LA

union(union(df2,df4),df6) #all ASV's that makes PA different
intersect(intersect(df2,df4),df6) #common ASV's that makes PA different

union(union(df3,df5),df6) #all ASV's that makes LA different
intersect(intersect(df3,df5),df6) #common ASV's that makes LA different

union(union(df1,df2),df3) #all ASV's that makes LS different
intersect(intersect(df1,df2),df3) #common ASV's that makes LS different

union(union(df1,df4),df5) #all ASV's that makes V different
intersect(intersect(df1,df4),df5) #common ASV's that makes V different

ASV_list <- union(df1,df2)
ASV_list <- union(ASV_list,df3)
ASV_list <- union(ASV_list,df4)
ASV_list <- union(ASV_list,df5)
ASV_list <- union(ASV_list,df6)

phy_mrg <- merge_samples(subset_samples(phy_norm, Sample == "Soil"), group = "Soil", fun = "mean")

phy_mrg <- prune_taxa(ASV_list, phy_mrg)

phy_mrg <- otu_table(t(phy_mrg))

phy_mrg <- as.data.frame(phy_mrg)

phy_mrg[,"ASV"] <- rownames(phy_mrg)

phy_mrg[,"taxonomy"] <- taxonomy2[rownames(taxonomy2) %in% phy_mrg$ASV,]

phy_mrg[,"Order"] <- rowMeans(phy_mrg[,1:4],)

phy_mrg <- phy_mrg[order(phy_mrg$Order,decreasing = TRUE),]
```

```{r echo= FALSE}
max(phy_mrg[,1:4])

quantile_breaks <- function(xs, n = 10) {
  breaks <- quantile(xs, probs = seq(0, 1, length.out = n))
  breaks[!duplicated(breaks)]
}

mat_breaks <- quantile_breaks(as.matrix(phy_mrg[,1:4]), n = 100)

colors = colorRampPalette(c("white","goldenrod1","orange","orangered", "darkred"))(length(mat_breaks)-1)

colnames(phy_mrg)

col_ann = data.frame(
  Soil=factor(c("La_Serena","Los_Angeles","Pozo_Almonte","Vicuna"
                ))
)

rownames(col_ann) = colnames(phy_mrg[,1:4])

pal = list(Soil = c("La_Serena" = "gold",
                    "Los_Angeles" = "orange",
                    "Vicuna" = "darkgoldenrod",
                    "Pozo_Almonte" = "brown")
           )
```

``` {r echo= FALSE}
pheatmap(phy_mrg[,c(1:4)], breaks=mat_breaks, col=colors, 
         angle_col = 45, cellwidth = 12,
         cluster_cols = FALSE, cluster_rows = FALSE, annotation_col = col_ann, 
         annotation_colors = pal, fontsize_col=8, fontsize_row=8,
         main = "Disimilarity fungal species", 
         labels_row = paste(phy_mrg$taxonomy,rownames(phy_mrg),sep = ",")
         , annotation_legend = FALSE
         , show_rownames = FALSE
         , na_col = 'gray'
         )
```

```{r}
tmp <- prune_samples(!is.na(sample_data(subset_samples(phy_norm, Sample == "Soil"))$Specie), subset_samples(phy_norm, Sample == "Soil")) 
adonis2(t(otu_table(tmp)) ~ Soil, data = data.frame(sample_data(subset_samples(phy_norm, Sample == "Soil"))))
```

```{r}
pairwise.adonis2(t(otu_table(tmp))~Soil, data = data.frame(sample_data(tmp)))
```

## Recruitment
#Data preparation
```{r}
phy_obj_F_Soil <- subset_samples(phy_obj_Fungi, Sample == "Soil")

phy_FS_V <- subset_samples(phy_obj_F_Soil, Soil == "Vicuna")

phy_FS_LS <- subset_samples(phy_obj_F_Soil, Soil == "La_Serena")

phy_FS_LA <- subset_samples(phy_obj_F_Soil, Soil == "Los_Angeles")
```

```{r}
phy_obj_F_Root <- subset_samples(phy_obj_Fungi, Sample == "Root")

phy_FR_VPc <- subset_samples(phy_obj_F_Soil, Soil == "Vicuna")
phy_FR_VPc <- subset_samples(phy_FR_VPc, Specie == "P_chilensis")

phy_FR_LSPc <- subset_samples(phy_obj_F_Soil, Soil == "La_Serena")
phy_FR_LSPc <- subset_samples(phy_FR_LSPc, Specie == "P_chilensis")

phy_FR_LAPc <- subset_samples(phy_obj_F_Soil, Soil == "Los_Angeles")
phy_FR_LAPc <- subset_samples(phy_FR_LAPc, Specie == "P_chilensis")

phy_FR_VPt <- subset_samples(phy_obj_F_Soil, Soil == "Vicuna")
phy_FR_VPt <- subset_samples(phy_FR_VPt, Specie == "P_tamarugo")

phy_FR_LSPt <- subset_samples(phy_obj_F_Soil, Soil == "La_Serena")
phy_FR_LSPt <- subset_samples(phy_FR_LSPt, Specie == "P_tamarugo")

phy_FR_LAPt <- subset_samples(phy_obj_F_Soil, Soil == "Los_Angeles")
phy_FR_LAPt <- subset_samples(phy_FR_LAPt, Specie == "P_tamarugo")
```

```{r}
phy_norm_F_Soil <- subset_samples(phy_norm, Sample == "Soil")

phy_norm_FS_mrg <- merge_samples(phy_norm_F_Soil, "Soil", fun = mean)

phy_norm_F_Root <- subset_samples(phy_norm, Sample == "Root")

sample_data(phy_norm_F_Root)[,"Soil_sp"] <- paste(sample_data(phy_norm_F_Root)$Soil, sample_data(phy_norm_F_Root)$Specie, sep = "_")

phy_norm_FR_mrg <- merge_samples(phy_norm_F_Root, "Soil_sp", fun = mean)
```

#VicuÃ±a
```{r}
common_V_VPc <- intersect(rownames(otu_table(phy_FS_V)), rownames(otu_table(phy_FR_VPc)))

common_V_VPt <- intersect(rownames(otu_table(phy_FS_V)), rownames(otu_table(phy_FR_VPt)))

Root_Pc_V <- t(otu_table(phy_norm_FR_mrg))[colnames(otu_table(phy_norm_FR_mrg)) %in% common_V_VPc]
Root_Pc_V <- as.data.frame(as.matrix(Root_Pc_V)) %>% dplyr::select(Vicuna_P_chilensis)

Soil_Pc_V <- t(otu_table(phy_norm_FS_mrg))[colnames(otu_table(phy_norm_FS_mrg)) %in% common_V_VPc]
Soil_Pc_V <- as.data.frame(as.matrix(Soil_Pc_V)) %>% dplyr::select(Vicuna)

Root_Pt_V <- t(otu_table(phy_norm_FR_mrg))[colnames(otu_table(phy_norm_FR_mrg)) %in% common_V_VPt]
Root_Pt_V <- as.data.frame(as.matrix(Root_Pt_V)) %>% dplyr::select(Vicuna_P_tamarugo)

Soil_Pt_V <- t(otu_table(phy_norm_FS_mrg))[colnames(otu_table(phy_norm_FS_mrg)) %in% common_V_VPt]
Soil_Pt_V <- as.data.frame(as.matrix(Soil_Pt_V)) %>% dplyr::select(Vicuna)

P_tam_v <- merge(Soil_Pt_V, Root_Pt_V, by= "row.names")
rownames(P_tam_v) <- P_tam_v$Row.names
P_tam_v[,"ASVs"] <- P_tam_v$Row.names
P_tam_v$ASVs <- as.factor(P_tam_v$ASVs)
P_tam_v[,"Delta"] <- P_tam_v$Vicuna_P_tamarugo-P_tam_v$Vicuna
P_tam_v <- P_tam_v[,-1]

P_tam_v <- merge(P_tam_v,tax_table(phy_norm_FR_mrg), by = "row.names")

ggplot(P_tam_v, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. tamarugo recruitment VicuÃ±a") +
  scale_y_discrete(labels=paste(P_tam_v$Kingdom,P_tam_v$Phylum,P_tam_v$Class,P_tam_v$Order,P_tam_v$Family,P_tam_v$Genus,P_tam_v$Specie,P_tam_v$ASVs))

length(P_tam_v$Delta[P_tam_v$Delta==0])
length(P_tam_v$Delta[P_tam_v$Delta>0])
length(P_tam_v$Delta[P_tam_v$Delta<0])

P_chil_v <- merge(Soil_Pc_V, Root_Pc_V, by= "row.names")
rownames(P_chil_v) <- P_chil_v$Row.names
P_chil_v[,"ASVs"] <- P_chil_v$Row.names
P_chil_v$ASVs <- as.factor(P_chil_v$ASVs)
P_chil_v[,"Delta"] <- P_chil_v$Vicuna_P_chilensis-P_chil_v$Vicuna
P_chil_v <- P_chil_v[,-1]

P_chil_v <- merge(P_chil_v,tax_table(phy_norm_FR_mrg), by = "row.names")

ggplot(P_chil_v, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. chilensis recruitment VicuÃ±a") +
  scale_y_discrete(labels=paste(P_chil_v$Kingdom,P_chil_v$Phylum,P_chil_v$Class,P_chil_v$Order,P_chil_v$Family,P_chil_v$Genus,P_chil_v$Specie,P_chil_v$ASVs))

length(P_chil_v$Delta[P_chil_v$Delta==0])
length(P_chil_v$Delta[P_chil_v$Delta>0])
length(P_chil_v$Delta[P_chil_v$Delta<0])

df1 <- P_tam_v[order(abs(P_tam_v$Delta),decreasing = TRUE),] %>% head(30)
df2 <- P_chil_v[order(abs(P_chil_v$Delta),decreasing = TRUE),] %>% head(30)

df3 <- union(df1$Row.names,df2$Row.names)
df1 <- P_tam_v[P_tam_v$Row.names %in% df3,]
df2 <- P_chil_v[P_chil_v$Row.names %in% df3,]

p1 <- ggplot(df2, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. chilensis recruitment VicuÃ±a") +
  scale_y_discrete(labels=paste(df2$Kingdom,df2$Phylum,df2$Class,df2$Order,df2$Family,df2$Genus,df2$Specie,df2$ASVs))

p2 <- ggplot(df1, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. tamarugo recruitment VicuÃ±a") +
  scale_y_discrete(labels=paste(df1$Kingdom,df1$Phylum,df1$Class,df1$Order,df1$Family,df1$Genus,df1$Specie,df1$ASVs))

grid.arrange(p1, p2, ncol=2)
```

## venn diagram
```{r}
#higher than zero
P_tam_V_hz <- P_tam_v %>% filter(Delta > 0)
P_tam_V_hz <- P_tam_V_hz$Row.names

P_chil_v_hz <- P_chil_v %>% filter(Delta > 0)
P_chil_v_hz <- P_chil_v_hz$Row.names

plot.new()
suppressWarnings(grid.draw(venn.diagram(
  list(P_tam_V_hz=P_tam_V_hz,P_chil_v_hz=P_chil_v_hz),
  filename=NULL, col=c("yellowgreen","green"), margin=c(0.35,0.35,0.35,0.35),
  category.names=c("P.tamarugo","P.chilensis"))))

#equal to zero
P_tam_V_ez <- P_tam_v %>% filter(Delta == 0)
P_tam_V_ez <- P_tam_V_ez$Row.names

P_chil_v_ez <- P_chil_v %>% filter(Delta == 0)
P_chil_v_ez <- P_chil_v_ez$Row.names

#lower than zero
P_tam_V_lz <- P_tam_v %>% filter(Delta < 0)
P_tam_V_lz <- P_tam_V_lz$Row.names

P_chil_v_lz <- P_chil_v %>% filter(Delta < 0)
P_chil_v_lz <- P_chil_v_lz$Row.names
```

###common recruitment
```{r}
common_V <- intersect(P_tam_V_hz, P_chil_v_hz)

common_V <- t(otu_table(phy_norm_FR_mrg))[colnames(otu_table(phy_norm_FR_mrg)) %in% common_V]

common_V <- as.data.frame(as.matrix(common_V))

common_V[,"Delta"] <- common_V$Vicuna_P_chilensis-common_V$Vicuna_P_tamarugo

common_V <- merge(common_V,tax_table(phy_norm_FR_mrg), by = "row.names")
rownames(common_V) <- common_V$Row.names
common_V <- common_V[,-1]
common_V[,"ASVs"] <- rownames(common_V)

ggplot(common_V, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-1, 1), low = "yellowgreen", mid = "gray", high = "green") +
  geom_point() + geom_point(size = 5.5) + xlim(-1, 1) +
  ggtitle("Common recruited VicuÃ±a") +
  scale_y_discrete(labels=paste(common_V$Kingdom,common_V$Phylum,common_V$Class,common_V$Order,common_V$Family,common_V$Genus,common_V$Specie,common_V$ASVs))

common_V2 <- common_V[order(abs(common_V$Delta),decreasing = TRUE),] %>% head(30)

ggplot(common_V2, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-1, 1), low = "yellowgreen", mid = "gray", high = "green") +
  geom_point() + geom_point(size = 5.5) + xlim(-1, 1) +
  ggtitle("Common recruited VicuÃ±a") +
  scale_y_discrete(labels=paste(common_V2$Kingdom,common_V2$Phylum,common_V2$Class,common_V2$Order,common_V2$Family,common_V2$Genus,common_V2$Specie,common_V2$ASVs))
```

#La Serena
```{r}
common_LS_LSPc <- intersect(rownames(otu_table(phy_FS_LS)), rownames(otu_table(phy_FR_LSPc)))

common_LS_LSPt <- intersect(rownames(otu_table(phy_FS_LS)), rownames(otu_table(phy_FR_LSPt)))

Root_Pc_LS <- t(otu_table(phy_norm_FR_mrg))[colnames(otu_table(phy_norm_FR_mrg)) %in% common_LS_LSPc]
Root_Pc_LS <- as.data.frame(as.matrix(Root_Pc_LS)) %>% dplyr::select(La_Serena_P_chilensis)

Soil_Pc_LS <- t(otu_table(phy_norm_FS_mrg))[colnames(otu_table(phy_norm_FS_mrg)) %in% common_LS_LSPc]
Soil_Pc_LS <- as.data.frame(as.matrix(Soil_Pc_LS)) %>% dplyr::select(La_Serena)

Root_Pt_LS <- t(otu_table(phy_norm_FR_mrg))[colnames(otu_table(phy_norm_FR_mrg)) %in% common_LS_LSPt]
Root_Pt_LS <- as.data.frame(as.matrix(Root_Pt_LS)) %>% dplyr::select(La_Serena_P_tamarugo)

Soil_Pt_LS <- t(otu_table(phy_norm_FS_mrg))[colnames(otu_table(phy_norm_FS_mrg)) %in% common_LS_LSPt]
Soil_Pt_LS <- as.data.frame(as.matrix(Soil_Pt_LS)) %>% dplyr::select(La_Serena)

P_tam_LS <- merge(Soil_Pt_LS, Root_Pt_LS, by= "row.names")
rownames(P_tam_LS) <- P_tam_LS$Row.names
P_tam_LS[,"ASVs"] <- P_tam_LS$Row.names
P_tam_LS$ASVs <- as.factor(P_tam_LS$ASVs)
P_tam_LS[,"Delta"] <- P_tam_LS$La_Serena_P_tamarugo-P_tam_LS$La_Serena
P_tam_LS <- P_tam_LS[,-1]

P_tam_LS <- merge(P_tam_LS,tax_table(phy_norm_FR_mrg), by = "row.names")

ggplot(P_tam_LS, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. tamarugo recruitment La Serena") +
  scale_y_discrete(labels=paste(P_tam_LS$Kingdom,P_tam_LS$Phylum,P_tam_LS$Class,P_tam_LS$Order,P_tam_LS$Family,P_tam_LS$Genus,P_tam_LS$Specie,P_tam_LS$ASVs))

length(P_tam_LS$Delta[P_tam_LS$Delta==0])
length(P_tam_LS$Delta[P_tam_LS$Delta>0])
length(P_tam_LS$Delta[P_tam_LS$Delta<0])

P_chil_LS <- merge(Soil_Pc_LS, Root_Pc_LS, by= "row.names")
rownames(P_chil_LS) <- P_chil_LS$Row.names
P_chil_LS[,"ASVs"] <- P_chil_LS$Row.names
P_chil_LS$ASVs <- as.factor(P_chil_LS$ASVs)
P_chil_LS[,"Delta"] <- P_chil_LS$La_Serena_P_chilensis-P_chil_LS$La_Serena
P_chil_LS <- P_chil_LS[,-1]

P_chil_LS <- merge(P_chil_LS,tax_table(phy_norm_FR_mrg), by = "row.names")

ggplot(P_chil_LS, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. chilensis recruitment La Serena") +
  scale_y_discrete(labels=paste(P_chil_LS$Kingdom,P_chil_LS$Phylum,P_chil_LS$Class,P_chil_LS$Order,P_chil_LS$Family,P_chil_LS$Genus,P_chil_LS$Specie,P_chil_LS$ASVs))

length(P_chil_LS$Delta[P_chil_LS$Delta==0])
length(P_chil_LS$Delta[P_chil_LS$Delta>0])
length(P_chil_LS$Delta[P_chil_LS$Delta<0])

df1 <- P_tam_LS[order(abs(P_tam_LS$Delta),decreasing = TRUE),] %>% head(30)
df2 <- P_chil_LS[order(abs(P_chil_LS$Delta),decreasing = TRUE),] %>% head(30)

df3 <- union(df1$Row.names,df2$Row.names)
df1 <- P_tam_LS[P_tam_LS$Row.names %in% df3,]
df2 <- P_chil_LS[P_chil_LS$Row.names %in% df3,]

p1 <- ggplot(df2, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. chilensis recruitment La Serena") +
  scale_y_discrete(labels=paste(df2$Kingdom,df2$Phylum,df2$Class,df2$Order,df2$Family,df2$Genus,df2$Specie,df2$ASVs))

p2 <- ggplot(df1, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. tamarugo recruitment La Serena") +
  scale_y_discrete(labels=paste(df1$Kingdom,df1$Phylum,df1$Class,df1$Order,df1$Family,df1$Genus,df1$Specie,df1$ASVs))

grid.arrange(p1, p2, ncol=2)
```

##venn diagram
```{r}
#higher than zero
P_tam_LS_hz <- P_tam_LS %>% filter(Delta > 0)
P_tam_LS_hz <- P_tam_LS_hz$Row.names

P_chil_LS_hz <- P_chil_LS %>% filter(Delta > 0)
P_chil_LS_hz <- P_chil_LS_hz$Row.names

plot.new()
suppressWarnings(grid.draw(venn.diagram(
  list(P_tam_LS_hz=P_tam_LS_hz,P_chil_LS_hz=P_chil_LS_hz),
  filename=NULL, col=c("yellowgreen","green"), margin=c(0.35,0.35,0.35,0.35),
  category.names=c("P.tamarugo","P.chilensis"))))

#equal to zero
P_tam_LS_ez <- P_tam_LS %>% filter(Delta == 0)
P_tam_LS_ez <- P_tam_LS_ez$Row.names

P_chil_LS_ez <- P_chil_LS %>% filter(Delta == 0)
P_chil_LS_ez <- P_chil_LS_ez$Row.names

#equal to zero
P_tam_LS_lz <- P_tam_LS %>% filter(Delta < 0)
P_tam_LS_lz <- P_tam_LS_lz$Row.names

P_chil_LS_lz <- P_chil_LS %>% filter(Delta < 0)
P_chil_LS_lz <- P_chil_LS_lz$Row.names
```

###common recruitment
```{r}
common_LS <- intersect(P_tam_LS_hz, P_chil_LS_hz)

common_LS <- t(otu_table(phy_norm_FR_mrg))[colnames(otu_table(phy_norm_FR_mrg)) %in% common_LS]

common_LS <- as.data.frame(as.matrix(common_LS))

common_LS[,"Delta"] <- common_LS$La_Serena_P_chilensis-common_LS$La_Serena_P_tamarugo

common_LS <- merge(common_LS,tax_table(phy_norm_FR_mrg), by = "row.names")
rownames(common_LS) <- common_LS$Row.names
common_LS <- common_LS[,-1]
common_LS[,"ASVs"] <- rownames(common_LS)

ggplot(common_LS, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-1, 1), low = "yellowgreen", mid = "gray", high = "green") +
  geom_point() + geom_point(size = 5.5) + xlim(-1, 1) +
  ggtitle("Common recruited VicuÃ±a") +
  scale_y_discrete(labels=paste(common_LS$Kingdom,common_LS$Phylum,common_LS$Class,common_LS$Order,common_LS$Family,common_LS$Genus,common_LS$Specie,common_LS$ASVs))

common_LS2 <- common_LS[order(abs(common_LS$Delta),decreasing = TRUE),] %>% head(30)

ggplot(common_LS2, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-1, 1), low = "yellowgreen", mid = "gray", high = "green") +
  geom_point() + geom_point(size = 5.5) + xlim(-1, 1) +
  ggtitle("Common recruited VicuÃ±a") +
  scale_y_discrete(labels=paste(common_LS2$Kingdom,common_LS2$Phylum,common_LS2$Class,common_LS2$Order,common_LS2$Family,common_LS2$Genus,common_LS2$Specie,common_LS2$ASVs))
```

#Los Angeles
```{r}
common_LA_LAPc <- intersect(rownames(otu_table(phy_FS_LA)), rownames(otu_table(phy_FR_LAPc)))

common_LA_LAPt <- intersect(rownames(otu_table(phy_FS_LA)), rownames(otu_table(phy_FR_LAPt)))

Root_Pc_LA <- t(otu_table(phy_norm_FR_mrg))[colnames(otu_table(phy_norm_FR_mrg)) %in% common_LA_LAPc]
Root_Pc_LA <- as.data.frame(as.matrix(Root_Pc_LA)) %>% dplyr::select(Los_Angeles_P_chilensis)

Soil_Pc_LA <- t(otu_table(phy_norm_FS_mrg))[colnames(otu_table(phy_norm_FS_mrg)) %in% common_LA_LAPc]
Soil_Pc_LA <- as.data.frame(as.matrix(Soil_Pc_LA)) %>% dplyr::select(Los_Angeles)

Root_Pt_LA <- t(otu_table(phy_norm_FR_mrg))[colnames(otu_table(phy_norm_FR_mrg)) %in% common_LA_LAPt]
Root_Pt_LA <- as.data.frame(as.matrix(Root_Pt_LA)) %>% dplyr::select(Los_Angeles_P_tamarugo)

Soil_Pt_LA <- t(otu_table(phy_norm_FS_mrg))[colnames(otu_table(phy_norm_FS_mrg)) %in% common_LA_LAPt]
Soil_Pt_LA <- as.data.frame(as.matrix(Soil_Pt_LA)) %>% dplyr::select(Los_Angeles)

P_tam_LA <- merge(Soil_Pt_LA, Root_Pt_LA, by= "row.names")
rownames(P_tam_LA) <- P_tam_LA$Row.names
P_tam_LA[,"ASVs"] <- P_tam_LA$Row.names
P_tam_LA$ASVs <- as.factor(P_tam_LA$ASVs)
P_tam_LA[,"Delta"] <- P_tam_LA$Los_Angeles_P_tamarugo-P_tam_LA$Los_Angeles
P_tam_LA <- P_tam_LA[,-1]

P_tam_LA <- merge(P_tam_LA,tax_table(phy_norm_FR_mrg), by = "row.names")

ggplot(P_tam_LA, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. tamarugo recruitment Los Angeles") +
  scale_y_discrete(labels=paste(P_tam_LA$Kingdom,P_tam_LA$Phylum,P_tam_LA$Class,P_tam_LA$Order,P_tam_LA$Family,P_tam_LA$Genus,P_tam_LA$Specie,P_tam_LA$ASVs))

length(P_tam_LA$Delta[P_tam_LA$Delta==0])
length(P_tam_LA$Delta[P_tam_LA$Delta>0])
length(P_tam_LA$Delta[P_tam_LA$Delta<0])

P_chil_LA <- merge(Soil_Pc_LA, Root_Pc_LA, by= "row.names")
rownames(P_chil_LA) <- P_chil_LA$Row.names
P_chil_LA[,"ASVs"] <- P_chil_LA$Row.names
P_chil_LA$ASVs <- as.factor(P_chil_LA$ASVs)
P_chil_LA[,"Delta"] <- P_chil_LA$Los_Angeles_P_chilensis-P_chil_LA$Los_Angeles
P_chil_LA <- P_chil_LA[,-1]

P_chil_LA <- merge(P_chil_LA,tax_table(phy_norm_FR_mrg), by = "row.names")

ggplot(P_chil_LA, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. chilensis recruitment Los Angeles") +
  scale_y_discrete(labels=paste(P_chil_LA$Kingdom,P_chil_LA$Phylum,P_chil_LA$Class,P_chil_LA$Order,P_chil_LA$Family,P_chil_LA$Genus,P_chil_LA$Specie,P_chil_LA$ASVs))

length(P_chil_LA$Delta[P_chil_LA$Delta==0])
length(P_chil_LA$Delta[P_chil_LA$Delta>0])
length(P_chil_LA$Delta[P_chil_LA$Delta<0])

df1 <- P_tam_LA[order(abs(P_tam_LA$Delta),decreasing = TRUE),] %>% head(30)
df2 <- P_chil_LA[order(abs(P_chil_LA$Delta),decreasing = TRUE),] %>% head(30)

df3 <- union(df1$Row.names,df2$Row.names)
df1 <- P_tam_LA[P_tam_LA$Row.names %in% df3,]
df2 <- P_chil_LA[P_chil_LA$Row.names %in% df3,]

p1 <- ggplot(df2, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. chilensis recruitment Los Angeles") +
  scale_y_discrete(labels=paste(df2$Kingdom,df2$Phylum,df2$Class,df2$Order,df2$Family,df2$Genus,df2$Specie,df2$ASVs))

p2 <- ggplot(df1, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. tamarugo recruitment Los Angeles") +
  scale_y_discrete(labels=paste(df1$Kingdom,df1$Phylum,df1$Class,df1$Order,df1$Family,df1$Genus,df1$Specie,df1$ASVs))

grid.arrange(p1, p2, ncol=2)
```

##venn diagram
```{r}
#higher than zero
P_tam_LA_hz <- P_tam_LA %>% filter(Delta > 0)
P_tam_LA_hz <- P_tam_LA_hz$Row.names

P_chil_LA_hz <- P_chil_LA %>% filter(Delta > 0)
P_chil_LA_hz <- P_chil_LA_hz$Row.names

#pdf("Manuscript/Plots/Supplementary/Venn_LA_fungi_hz.pdf", width = 10, height = 10)
plot.new()
suppressWarnings(grid.draw(venn.diagram(
  list(P_tam_LA_hz=P_tam_LA_hz,P_chil_LA_hz=P_chil_LA_hz),
  filename=NULL, col=c("yellowgreen","green"), margin=c(0.35,0.35,0.35,0.35),
  category.names=c("P.tamarugo","P.chilensis"))))
#dev.off()

#equal to zero
P_tam_LA_ez <- P_tam_LA %>% filter(Delta == 0)
P_tam_LA_ez <- P_tam_LA_ez$Row.names

P_chil_LA_ez <- P_chil_LA %>% filter(Delta == 0)
P_chil_LA_ez <- P_chil_LA_ez$Row.names

#lower than zero
P_tam_LA_lz <- P_tam_LA %>% filter(Delta < 0)
P_tam_LA_lz <- P_tam_LA_lz$Row.names

P_chil_LA_lz <- P_chil_LA %>% filter(Delta < 0)
P_chil_LA_lz <- P_chil_LA_lz$Row.names
```

###common recruitment
```{r}
common_LA <- intersect(P_tam_LA_hz, P_chil_LA_hz)

common_LA <- t(otu_table(phy_norm_FR_mrg))[colnames(otu_table(phy_norm_FR_mrg)) %in% common_LA]

common_LA <- as.data.frame(as.matrix(common_LA))

common_LA[,"Delta"] <- common_LA$Los_Angeles_P_chilensis-common_LA$Los_Angeles_P_tamarugo

common_LA <- merge(common_LA,tax_table(phy_norm_FR_mrg), by = "row.names")
rownames(common_LA) <- common_LA$Row.names
common_LA <- common_LA[,-1]
common_LA[,"ASVs"] <- rownames(common_LA)

ggplot(common_LA, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-1, 1), low = "yellowgreen", mid = "gray", high = "green") +
  geom_point() + geom_point(size = 5.5) + xlim(-1, 1) +
  ggtitle("Common recruited VicuÃ±a") +
  scale_y_discrete(labels=paste(common_LA$Kingdom,common_LA$Phylum,common_LA$Class,common_LA$Order,common_LA$Family,common_LA$Genus,common_LA$Specie,common_LA$ASVs))

common_LA2 <- common_LA[order(abs(common_LA$Delta),decreasing = TRUE),] %>% head(30)

ggplot(common_LA2, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-1, 1), low = "yellowgreen", mid = "gray", high = "green") +
  geom_point() + geom_point(size = 5.5) + xlim(-1, 1) +
  ggtitle("Common recruited VicuÃ±a") +
  scale_y_discrete(labels=paste(common_LA2$Kingdom,common_LA2$Phylum,common_LA2$Class,common_LA2$Order,common_LA2$Family,common_LA2$Genus,common_LA2$Specie,common_LA2$ASVs))
```

#Bacterial metagenonics
## Amplicon sequencing analysis

The amplicon sequencing analysis was made using qiime2 for pre-processing and phyloseq package for the phylogenetic analysis.

phyloseq object preparation
```{r}
# ASV table
ASVs <- read.table("16S/table_16S.txt", sep="\t", header=TRUE, row.names=1)
# taxonomy
taxonomy <- read.table("16S/taxonomy.tsv", sep="\t", header=TRUE, row.names=1, na.strings=c(""," ","NA"))
taxonomy <- as.matrix(taxonomy)
# tree
tree <- read_tree("16S/tree.nwk")
# sample data
metadata <- read.table("16S/metadata.tsv", sep="\t", header=TRUE, row.names=1, na.strings=c(""," ","NA"),dec = ",")
# phyloseq object
phy_obj <- phyloseq(otu_table(ASVs, taxa_are_rows = TRUE), 
                    tax_table(taxonomy), sample_data(metadata),tree)

tax_table(phy_obj) <- gsub(".__","",tax_table(phy_obj)) #separate taxonomy in columns

phy_obj_Bacteria <- phy_obj

phy_obj_Bacteria <- subset_taxa(phy_obj, Kingdom == "Bacteria") #keep just bacteria
phy_obj_Bacteria

phy_obj_Bacteria <- subset_taxa(phy_obj_Bacteria, Family != "Mitochondria") #remove mitochondria family
phy_obj_Bacteria
```

Filtering ASV equal or lower than 10. Filtering was made with filterfun() from the genefilter package to set the threshold and filter_taxa from phyloseq to remove the ASV lower than the set filter #revise
```{r}
flist    <- filterfun(kOverA(1, 10)) #remove AVS's equal or lower than 10 
phy_obj_Bacteria <- filter_taxa(phy_obj_Bacteria, flist, TRUE)
phy_obj_Bacteria
sample_data(phy_obj_Bacteria)
```

Filtering ASV with a minimum abundance of 0.005% per sample.
First the 0.005% had to be calculated for each sample type, by merging the phyloseq object by sample type (merge_samples()), we get the total number of reads (sample_sums()) and we can calculate the 0.005% for each. 
Then, the phyloseq object has to be splitted by sample type (subset_samples()) and the ASVs lower than the calculated 0.005% (abundance object) can be removed (prune_taxa()).
Finally, the phyloseq objects by sample type can be merged into one new phyloseq object (merge_phyloseq())
```{r}
Pch <- subset_samples(phy_obj_Bacteria,Specie=="P_chilensis")

abundance <- as.data.frame(sample_sums(merge_samples(Pch,group = "Sample")))
colnames(abundance) <- "abundance"
abundance[,".005%"] <- abundance$abundance*.00005
abundance$`.005%` <- round(abundance$`.005%`)
abundance #minimum number of reads to remove per-sample
abundance <- t(abundance)[-1,]
abundance <- as.data.frame(t(abundance))
abundance

#Subseting by sample type
Root <- subset_samples(Pch,Sample=="Root")
Root
Soil <- subset_samples(Pch,Sample=="Soil")
Soil

#Removing ASV lower than 0.005% of abundance per sample type

Root <- prune_taxa(taxa_sums(Root) > abundance$Root,Root)
Root
Soil <- prune_taxa(taxa_sums(Soil) > abundance$Soil,Soil)
Soil

#preparing for merge into a new phy_obj 
Root <- phyloseq(otu_table(Root),sample_data(Root),tax_table(Root))
Soil <- phyloseq(otu_table(Soil),sample_data(Soil),tax_table(Soil))

Pch <- merge_phyloseq(Root,Soil)
Pch
```

```{r}
Pta <- subset_samples(phy_obj_Bacteria,Specie=="P_tamarugo")

abundance <- as.data.frame(sample_sums(merge_samples(Pta,group = "Sample")))
colnames(abundance) <- "abundance"
abundance[,".005%"] <- abundance$abundance*.00005
abundance$`.005%` <- round(abundance$`.005%`)
abundance #minimum number of reads to remove per-sample
abundance <- t(abundance)[-1,]
abundance <- as.data.frame(t(abundance))
abundance

#Subseting by sample type
Root <- subset_samples(Pta,Sample=="Root")
Root
Soil <- subset_samples(Pta,Sample=="Soil")
Soil

#Removing ASV lower than 0.005% of abundance per sample type

Root <- prune_taxa(taxa_sums(Root) > abundance$Root,Root)
Root
Soil <- prune_taxa(taxa_sums(Soil) > abundance$Soil,Soil)
Soil

#preparing for merge into a new phy_obj_Bacteria 
Root <- phyloseq(otu_table(Root),sample_data(Root),tax_table(Root))
Soil <- phyloseq(otu_table(Soil),sample_data(Soil),tax_table(Soil))

Pta <- merge_phyloseq(Root,Soil)
Pta
```

```{r}
#Adding phy_tree
phy_obj_Bacteria <- merge_phyloseq(Pch,Pta)
phy_obj_Bacteria <- phyloseq(otu_table(phy_obj_Bacteria),sample_data(phy_obj_Bacteria),tax_table(phy_obj_Bacteria),tree)
phy_obj_Bacteria
```

Merging phylogenetic tree tips.
This is made to avoid many ASVs annotating for the same specie. Using the tip_glom() function, the tips of the phylogenetic tree lower than a given 'height' will be merged into one.
```{r}
phy_obj_Bacteria <- tip_glom(phy_obj_Bacteria,hcfun = hclust)
phy_obj_Bacteria
```

Library size and normalization
```{r}
#Library size
sums <- sample_sums(phy_obj_Bacteria)
barplot(sums[order(sums, decreasing=TRUE)], las=3, cex.names = 0.5, ylab="Library size", col = "skyblue")
abline(h = 10000, lty=2)

#Normalisation by proportions
phy_obj_Bacteria <- prune_samples(sample_sums(phy_obj_Bacteria)>10000,phy_obj_Bacteria)
phy_obj_Bacteria

phy_norm <- transform_sample_counts(phy_obj_Bacteria,function(x) x/sum(x))
barplot(sample_sums(phy_norm),las=3,cex.names=.5,col="skyblue")

#Rarefy
set.seed(12345)
phy_raref <- rarefy_even_depth(phy_obj_Bacteria, sample.size = min(sample_sums(phy_obj_Bacteria)), replace = FALSE, trimOTUs = TRUE, verbose = TRUE)
barplot(sample_sums(phy_raref), las=3, cex.names = 0.5, col = "skyblue")
```

## Richness, alpha and beta diversity
### Richness 
Richness was calculated using the pd() function from picante package.
Preparing richness dataframe
```{r}
faiths_div <- pd(as.data.frame(t(otu_table(phy_raref))), phy_tree(phy_obj_Bacteria), include.root = FALSE)
faiths_div <- merge(faiths_div,sample_data(phy_obj_Bacteria), by = "row.names")
faiths_div$Soil <- factor(faiths_div$Soil, levels=c('Pozo_Almonte','La_Serena','Vicuna','Los_Angeles'))
faiths_div$Specie <- as.factor(faiths_div$Specie)
```

Plotting richness
```{r echo= FALSE}
#pdf("Manuscript/Plots/Bacterial_Richness_soil.pdf")
ggplot(data=faiths_div %>% filter(Sample=="Soil"), aes(x = Sample, y = SR, fill = Soil)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("brown", "gold","darkgoldenrod3","darkorange")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "Specie richness")
#dev.off()
```

```{r echo= FALSE}
#pdf("Manuscript/Plots/Bacterial_Richness.pdf")
ggplot(faiths_div, aes(x = Specie, y = SR, fill = Soil)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("brown", "gold","darkgoldenrod3","darkorange")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "Specie richness",limits = c(0,300)) + 
  facet_grid(.~Sample)
#dev.off()
```
Statistical analysis of richness using Kruskal-Wallis. Any significant differences were further analysed with Dunnâ€™s test with dunnTest() function from FAS package.
The model used was richness~Sample type, richness~ fertilization type and richness~sample*fertilization
```{r}
r_root <- faiths_div %>%
  filter(Sample == "Root")

agricolae::kruskal(r_root$SR, r_root$Specie, p.adj = "fdr", group = T, console = T)

agricolae::kruskal(r_root %>% filter(Specie=="P_chilensis") %>% dplyr::select(SR), r_root %>% filter(Specie=="P_chilensis") %>% dplyr::select(Soil), p.adj = "BH", group = T, console = T)

agricolae::kruskal(r_root %>% filter(Specie=="P_tamarugo") %>% dplyr::select(SR), r_root %>% filter(Specie=="P_tamarugo") %>% dplyr::select(Soil), p.adj = "fdr", group = T, console = T)

r_soil <- faiths_div %>%
  filter(Sample == "Soil")

agricolae::kruskal(r_soil$SR, r_soil$Soil, p.adj = "fdr", group = T, console = T)
```

### Alpha diversity

The Alpha diversity was calculated using the Shannon diversity index with the function diversity() from the vegan package. 

Calculating Shannon diversity
```{r}
shannon_div <- vegan::diversity(otu_table(phy_obj_Bacteria), MARGIN = 2, index = "shannon")
shannon_div <- as.data.frame(shannon_div)
shannon_div <- merge(shannon_div,sample_data(phy_obj_Bacteria), by = "row.names")
shannon_div$Soil <- factor(shannon_div$Soil, levels=c('Pozo_Almonte','La_Serena','Vicuna','Los_Angeles'))
shannon_div$Specie <- as.factor(shannon_div$Specie) 
```

Plotting alpha diversity index
```{r echo= FALSE}
#pdf("Manuscript/Plots/Bacterial_alpha_soil.pdf")
ggplot(data=shannon_div %>% filter(Sample=="Soil"), aes(x = Treatment, y = shannon_div, fill = Soil)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("brown", "gold","darkgoldenrod3","darkorange")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "Shannon Diversity Index")
#dev.off()
```

```{r echo= FALSE}
#pdf("Manuscript/Plots/Bacterial_alpha.pdf")
ggplot(shannon_div, aes(x = Specie, y = shannon_div, fill = Soil)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("brown", "gold","darkgoldenrod3","darkorange")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "Shannon Diversity Index",limits = c(0,6)) + 
  facet_grid(.~Sample)
#dev.off()
```

Statistical analysis of Shannon diversity index using Kruskal-Wallis. Any significant differences were further analysed with Dunnâ€™s test with dunnTest() function from FAS package.
The model used was shannon~Sample type, shannon~fertilization type and shannon~sample*fertilization
```{r}
a_root <- shannon_div %>%
  filter(Sample == "Root")

agricolae::kruskal(a_root$shannon_div, a_root$Specie, p.adj = "fdr", group = T, console = T)

agricolae::kruskal(a_root %>% filter(Specie=="P_chilensis") %>% dplyr::select(shannon_div), a_root %>% filter(Specie=="P_chilensis") %>% dplyr::select(Soil), p.adj = "fdr", group = T, console = T)

agricolae::kruskal(a_root %>% filter(Specie=="P_tamarugo") %>% dplyr::select(shannon_div), a_root %>% filter(Specie=="P_tamarugo") %>% dplyr::select(Soil), p.adj = "fdr", group = T, console = T)

a_soil <- shannon_div %>%
  filter(Sample == "Soil")

agricolae::kruskal(a_soil$shannon_div, a_soil$Soil, p.adj = "fdr", group = T, console = T)
```

### Evenness
```{r}
even <- otu_table(phy_obj_Bacteria)
even <- as.data.frame(even)

tax <- paste(tax_table(phy_obj_Bacteria)[,1],tax_table(phy_obj_Bacteria)[,2],tax_table(phy_obj_Bacteria)[,3],
      tax_table(phy_obj_Bacteria)[,4],tax_table(phy_obj_Bacteria)[,5],tax_table(phy_obj_Bacteria)[,6],
      tax_table(phy_obj_Bacteria)[,7], sep = "; ")
even[,"taxonomy"] <- tax

even <- RAM::evenness(data=list(even=even), index = "shannon")
even <- as.data.frame(t(even))

even <- merge(even,sample_data(phy_obj_Bacteria), by = "row.names")
even$Soil <- factor(even$Soil, levels=c('Pozo_Almonte','La_Serena','Vicuna','Los_Angeles'))
```

```{r}
#pdf("Manuscript/Plots/Bacterial_even_soil.pdf")
ggplot(data=even %>% filter(Sample=="Soil"), aes(x = Treatment, y = even, fill = Soil)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("brown", "gold","darkgoldenrod3","darkorange")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "Pielou's Evenness")
#dev.off()
```

```{r}
#pdf("Manuscript/Plots/Bacterial_even.pdf")
ggplot(even, aes(x = Specie, y = even, fill = Soil)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("brown", "gold","darkgoldenrod3","darkorange")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "Pielou's Evennes",limits = c(0,1)) + 
  facet_grid(.~Sample)
#dev.off()
```

```{r}
e_root <- even %>%
  filter(Sample == "Root")

agricolae::kruskal(e_root$even, e_root$Specie, p.adj = "fdr", group = T, console = T)

agricolae::kruskal(e_root %>% filter(Specie=="P_chilensis") %>% dplyr::select(even), e_root %>% filter(Specie=="P_chilensis") %>% dplyr::select(Soil), p.adj = "fdr", group = T, console = T)

agricolae::kruskal(e_root %>% filter(Specie=="P_tamarugo") %>% dplyr::select(even), e_root %>% filter(Specie=="P_tamarugo") %>% dplyr::select(Soil), p.adj = "fdr", group = T, console = T)

e_soil <- even %>%
  filter(Sample == "Soil")

agricolae::kruskal(e_soil$even, e_soil$Soil, p.adj = "fdr", group = T, console = T)
```

```{r}
df <- faiths_div %>%
  left_join(
    shannon_div, by = "Row.names") %>%
  left_join(
    even, by = "Row.names")


df <- df %>% dplyr::select("Row.names", "SR","shannon_div","even","Sample","Specie","Soil")
df[,"all"] <- paste(df$Sample,df$Soil, sep = ".")

df <- as.data.frame(c(aggregate(SR~all, FUN=function(x) c(mean = mean(x), sd = sd(x)), data = df),(aggregate(shannon_div~all, FUN=function(x) c(mean = mean(x), sd = sd(x)), data = df)),(aggregate(even~all, FUN=function(x) c(mean = mean(x), sd = sd(x)), data = df))))

df <- df %>% dplyr::select(all,SR.mean, SR.sd, shannon_div.mean, shannon_div.sd, even.mean, even.sd)

rownames(df) <- df$all
df <- df[-1]

df$SR.mean <- round(df$SR.mean, digits = 2)
df$shannon_div.mean <- round(df$shannon_div.mean, digits = 2)
df$even.mean <- round(df$even.mean, digits = 2)

df$SR.sd <- round(df$SR.sd, digits = 3)
df$shannon_div.sd <- round(df$shannon_div.sd, digits = 3)
df$even.sd <- round(df$even.sd, digits = 3)
df

#write.csv(df,"Manuscript/Plots/Supplementary/Bacterial_soil_div.csv")
```


```{r}
Pch <- faiths_div %>%
  filter(Specie=="P_chilensis") %>%
  left_join(
    shannon_div, by = "Row.names") %>%
  left_join(
    even, by = "Row.names")

Pch <- Pch %>% dplyr::select("Row.names", "SR","shannon_div","even","Sample","Specie","Soil")
Pch[,"all"] <- paste(Pch$Sample,Pch$Specie,Pch$Soil, sep = ".")

df <- as.data.frame(c(aggregate(SR~all, FUN=function(x) c(mean = mean(x), sd = sd(x)), data = Pch),(aggregate(shannon_div~all, FUN=function(x) c(mean = mean(x), sd = sd(x)), data = Pch)),(aggregate(even~all, FUN=function(x) c(mean = mean(x), sd = sd(x)), data = Pch))))

df <- df %>% dplyr::select(all,SR.mean, SR.sd, shannon_div.mean, shannon_div.sd, even.mean, even.sd)

rownames(df) <- df$all
df <- df[-1]

Pta <- faiths_div %>%
  filter(Specie=="P_tamarugo") %>%
  left_join(
    shannon_div, by = "Row.names") %>%
  left_join(
    even, by = "Row.names")

Pta <- Pta %>% dplyr::select("Row.names", "SR","shannon_div","even","Sample","Specie","Soil")
Pta[,"all"] <- paste(Pta$Sample,Pta$Specie,Pta$Soil, sep = ".")

df2 <- as.data.frame(c(aggregate(SR~all, FUN=function(x) c(mean = mean(x), sd = sd(x)), data = Pta),(aggregate(shannon_div~all, FUN=function(x) c(mean = mean(x), sd = sd(x)), data = Pta)),(aggregate(even~all, FUN=function(x) c(mean = mean(x), sd = sd(x)), data = Pta))))

df2 <- df2 %>% dplyr::select(all,SR.mean, SR.sd, shannon_div.mean, shannon_div.sd, even.mean, even.sd)

rownames(df2) <- df2$all
df2 <- df2[-1]

df <- rbind(df,df2)

df$SR.mean <- round(df$SR.mean, digits = 2)
df$shannon_div.mean <- round(df$shannon_div.mean, digits = 2)
df$even.mean <- round(df$even.mean, digits = 2)

df$SR.sd <- round(df$SR.sd, digits = 3)
df$shannon_div.sd <- round(df$shannon_div.sd, digits = 3)
df$even.sd <- round(df$even.sd, digits = 3)

df

#write.csv(df,"Manuscript/Plots/Supplementary/Bacterial_richness_alpha.csv")
```

### Beta diversity
Beta diversity was evaluated using PCoA with bray-curtis distance

```{r echo= FALSE}
#Soil
ord <- ordinate(subset_samples(phy_raref, Sample == "Soil"), method = "PCoA", distance = "bray", weighted=FALSE)
sample_data(phy_norm)[,"Specie_sample"] <- paste(sample_data(phy_norm)$Specie,sample_data(phy_norm)$Sample, sep="-")

plot1<- gg_ordiplot(ord$vectors, groups = sample_data(subset_samples(phy_norm, Sample == "Soil"))$Soil, hull = TRUE, 
            label = TRUE, spiders = TRUE, ellipse = FALSE, pt.size = 7, plot = FALSE)

ggplot(data=plot1$df_spiders) + 
  geom_point(aes(x=x, y=y,shape=Group, colour=Group), size = 5) +
  geom_point(aes(x=cntr.x, y=cntr.y), size = 5, shape = 18) +
  scale_color_manual(values=c("gold", "darkorange", "brown", "darkgoldenrod3")) + scale_shape_manual(values = c(19,19,19,19)) +
  theme_bw() + 
  theme(legend.position = "bottom", legend.box = "vertical", 
        text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        panel.border = element_rect(size = 2, fill = NA))
```

```{r echo= FALSE}
phy_obj_B_Soil <- subset_samples(phy_obj_Bacteria, Sample == "Soil")
phy_obj_B_Soil <- subset_samples(phy_obj_B_Soil, Soil != "Pozo_Almonte")

sample_data(phy_obj_B_Soil)[,"sample_site"] <- paste(sample_data(phy_obj_B_Soil)$Sample, sample_data(phy_obj_B_Soil)$Soil, sep = "_")

phy_norm_B_Soil <- subset_samples(phy_norm, Sample == "Soil")
phy_norm_B_Soil <- subset_samples(phy_norm_B_Soil, Soil != "Pozo_Almonte")

sample_data(phy_norm_B_Soil)[,"sample_site"] <- paste(sample_data(phy_norm_B_Soil)$Sample, sample_data(phy_norm_B_Soil)$Soil, sep = "_")

phy_raref_B_Soil <- subset_samples(phy_raref, Sample == "Soil")
phy_raref_B_Soil <- subset_samples(phy_raref_B_Soil, Soil != "Pozo_Almonte")

sample_data(phy_raref_B_Soil)[,"sample_site"] <- paste(sample_data(phy_raref_B_Soil)$Sample, sample_data(phy_raref_B_Soil)$Soil, sep = "_")

#Root
phy_obj_B_Root <- subset_samples(phy_obj_Bacteria, Sample == "Root")
phy_norm_B_Root <- subset_samples(phy_norm, Sample == "Root")
phy_raref_B_Root <- subset_samples(phy_raref, Sample == "Root")

sample_data(phy_obj_B_Root)[,"sample_site"] <- paste(sample_data(phy_obj_B_Root)$Sample, sample_data(phy_obj_B_Root)$Soil, sample_data(phy_obj_B_Root)$Specie, sep = "_")

sample_data(phy_norm_B_Root)[,"sample_site"] <- paste(sample_data(phy_norm_B_Root)$Sample, sample_data(phy_norm_B_Root)$Soil, sample_data(phy_norm_B_Root)$Specie, sep = "_")

sample_data(phy_raref_B_Root)[,"sample_site"] <- paste(sample_data(phy_raref_B_Root)$Sample, sample_data(phy_raref_B_Root)$Soil, sample_data(phy_raref_B_Root)$Specie, sep = "_")

phy_obj2 <- merge_phyloseq(phy_obj_B_Soil,phy_obj_B_Root)

phy_norm2 <- merge_phyloseq(phy_norm_B_Soil,phy_norm_B_Root)

phy_raref2 <- merge_phyloseq(phy_raref_B_Soil,phy_raref_B_Root)

ord <- ordinate(phy_raref2, method = "PCoA", distance = "bray", weighted=FALSE)

plot_ordination(phy_norm2, ord, color = "Soil", shape = "sample_site") + geom_point(size=7) +
  scale_color_manual(values=c("gold", "darkorange", "darkgoldenrod3")) + scale_shape_manual(values = c(24,25,24,25,24,25,19,19,19,19)) +
  theme_bw() + 
  theme(legend.position = "bottom", legend.box = "vertical", 
        text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        panel.border = element_rect(size = 2, fill = NA))

plot1<- gg_ordiplot(ord$vectors, groups = sample_data(phy_norm2)$sample_site, hull = TRUE, 
            label = TRUE, spiders = TRUE, ellipse = FALSE, pt.size = 7, plot = FALSE)

#pdf("Manuscript/Plots/Fungal_beta_Both.pdf",height=10, width=10)
ggplot(data=plot1$df_spiders) + 
  geom_point(aes(x=x, y=y,shape=Group, colour=Group), size = 5) +
  geom_point(aes(x=cntr.x, y=cntr.y), size = 5, shape = 18) +
  scale_color_manual(values=c("gold","gold", "darkorange","darkorange", "darkgoldenrod3","darkgoldenrod3","gold", "darkorange", "darkgoldenrod3")) + scale_shape_manual(values = c(24,25,24,25,24,25,19,19,19,19)) +
  theme_bw() + 
  theme(legend.position = "bottom", legend.box = "vertical", 
        text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        panel.border = element_rect(size = 2, fill = NA))
#dev.off()

(sim <- with(sample_data(subset_samples(phy_obj2, Sample == "Root")), simper(t(otu_table(subset_samples(phy_obj2, Sample == "Root"))), sample_site)))
#summary(sim)
```

Statistical analysis of beta diversity using PerMANOVA with adonis() function from vegan package.
The model used was ASVs~Sample*fertilization
```{r}
tmp <- prune_samples(!is.na(sample_data(phy_obj2)$Specie), phy_obj2) 
adonis(t(otu_table(tmp)) ~ Specie * Sample * Soil, data = data.frame(sample_data(phy_obj2)))
#pairwise.adonis(t(otu_table(tmp)),data.frame(sample_data(phy_obj2))$Specie)
#pairwise.adonis(t(otu_table(tmp)),data.frame(sample_data(phy_obj2))$Sample)
#pairwise.adonis(t(otu_table(tmp)),data.frame(sample_data(phy_obj2))$Soil)
#pairwise.adonis(t(otu_table(tmp)),paste(data.frame(sample_data(phy_obj2))$Specie,data.frame(sample_data(phy_obj2))$Sample,sep = "."))
#pairwise.adonis(t(otu_table(tmp)),paste(data.frame(sample_data(phy_obj2))$Specie,data.frame(sample_data(phy_obj2))$Soil,sep = "."))
#pairwise.adonis(t(otu_table(tmp)),paste(data.frame(sample_data(phy_obj2))$Specie,data.frame(sample_data(phy_obj2))$Sample, data.frame(sample_data(phy_obj2))$Soil,sep = "."))
```

## Community composition
### Disimilarity Heatmap
``` {r echo= FALSE}
# Preparing data

taxonomy2 <- data.frame(paste(tax_table(subset_samples(phy_norm, Sample == "Soil"))[,1],tax_table(subset_samples(phy_norm, Sample == "Soil"))[,2],tax_table(subset_samples(phy_norm, Sample == "Soil"))[,3],tax_table(subset_samples(phy_norm, Sample == "Soil"))[,4],
                  tax_table(subset_samples(phy_norm, Sample == "Soil"))[,5],tax_table(subset_samples(phy_norm, Sample == "Soil"))[,6],tax_table(subset_samples(phy_norm, Sample == "Soil"))[,7],sep = ";"))
rownames(taxonomy2) <- rownames(tax_table(subset_samples(phy_norm, Sample == "Soil")))
colnames(taxonomy2) <- "taxonomy"
taxonomy2[,"ASV"] <- rownames(taxonomy2)

sim <- with(sample_data(subset_samples(phy_obj_Bacteria, Sample == "Soil")), simper(t(otu_table(subset_samples(phy_obj_Bacteria, Sample == "Soil"))), Soil))
#summary(sim)

df1 <- rownames(summary(sim)[[1]] %>% filter(cumsum < .7)) #LS_V
df2 <- rownames(summary(sim)[[2]] %>% filter(cumsum < .7)) #LS_PA
df3 <- rownames(summary(sim)[[3]] %>% filter(cumsum < .7)) #LS_LA
df4 <- rownames(summary(sim)[[4]] %>% filter(cumsum < .7)) #V_PA
df5 <- rownames(summary(sim)[[5]] %>% filter(cumsum < .7)) #V_LA
df6 <- rownames(summary(sim)[[6]] %>% filter(cumsum < .7)) #PA_LA

union(union(df2,df4),df6) #all ASV's that makes PA different
intersect(intersect(df2,df4),df6) #common ASV's that makes PA different

union(union(df3,df5),df6) #all ASV's that makes LA different
intersect(intersect(df3,df5),df6) #common ASV's that makes LA different

union(union(df1,df2),df3) #all ASV's that makes LS different
intersect(intersect(df1,df2),df3) #common ASV's that makes LS different

union(union(df1,df4),df5) #all ASV's that makes V different
intersect(intersect(df1,df4),df5) #common ASV's that makes V different

ASV_list <- union(df1,df2)
ASV_list <- union(ASV_list,df3)
ASV_list <- union(ASV_list,df4)
ASV_list <- union(ASV_list,df5)
ASV_list <- union(ASV_list,df6)

phy_mrg <- merge_samples(subset_samples(phy_norm, Sample == "Soil"), group = "Soil", fun = "mean")

phy_mrg <- prune_taxa(ASV_list, phy_mrg)

phy_mrg <- otu_table(t(phy_mrg))

phy_mrg <- as.data.frame(phy_mrg)

phy_mrg[,"ASV"] <- rownames(phy_mrg)

phy_mrg[,"taxonomy"] <- taxonomy2[rownames(taxonomy2) %in% phy_mrg$ASV,]

phy_mrg[,"Order"] <- rowMeans(phy_mrg[,1:4],)

phy_mrg <- phy_mrg[order(phy_mrg$Order,decreasing = TRUE),]
```

```{r echo= FALSE}
max(phy_mrg[,1:4])

quantile_breaks <- function(xs, n = 10) {
  breaks <- quantile(xs, probs = seq(0, 1, length.out = n))
  breaks[!duplicated(breaks)]
}

mat_breaks <- quantile_breaks(as.matrix(phy_mrg[,1:4]), n = 100)

colors = colorRampPalette(c("white","goldenrod1","orange","orangered", "darkred"))(length(mat_breaks)-1)

colnames(phy_mrg)

col_ann = data.frame(
  Soil=factor(c("La_Serena","Los_Angeles","Pozo_Almonte","Vicuna"
                ))
)

rownames(col_ann) = colnames(phy_mrg[,1:4])

pal = list(Soil = c("La_Serena" = "gold",
                    "Los_Angeles" = "orange",
                    "Vicuna" = "darkgoldenrod",
                    "Pozo_Almonte" = "brown")
           )
```

``` {r echo= FALSE}
pheatmap(phy_mrg[,c(1:4)], breaks=mat_breaks, col=colors, 
         angle_col = 45, cellwidth = 12,
         cluster_cols = FALSE, cluster_rows = FALSE, annotation_col = col_ann, 
         annotation_colors = pal, fontsize_col=8, fontsize_row=8,
         main = "Disimilarity Bacterial species", 
         labels_row = paste(phy_mrg$taxonomy,rownames(phy_mrg),sep = ",")
         , annotation_legend = FALSE
         , show_rownames = FALSE
         , na_col = 'gray'
         )
```

```{r}
tmp <- prune_samples(!is.na(sample_data(subset_samples(phy_norm, Sample == "Soil"))$Specie), subset_samples(phy_norm, Sample == "Soil")) 
adonis2(t(otu_table(tmp)) ~ Soil, data = data.frame(sample_data(subset_samples(phy_norm, Sample == "Soil"))))
```

```{r}
pairwise.adonis2(t(otu_table(tmp))~Soil, data = data.frame(sample_data(tmp)))
```

## Recruitment
#Data preparation
```{r}
phy_obj_B_Soil <- subset_samples(phy_obj_Bacteria, Sample == "Soil")

phy_BS_V <- subset_samples(phy_obj_B_Soil, Soil == "Vicuna")

phy_BS_LS <- subset_samples(phy_obj_B_Soil, Soil == "La_Serena")

phy_BS_LA <- subset_samples(phy_obj_B_Soil, Soil == "Los_Angeles")
```

```{r}
phy_obj_B_Root <- subset_samples(phy_obj_Bacteria, Sample == "Root")

phy_BR_VPc <- subset_samples(phy_obj_B_Soil, Soil == "Vicuna")
phy_BR_VPc <- subset_samples(phy_BR_VPc, Specie == "P_chilensis")

phy_BR_LSPc <- subset_samples(phy_obj_B_Soil, Soil == "La_Serena")
phy_BR_LSPc <- subset_samples(phy_BR_LSPc, Specie == "P_chilensis")

phy_BR_LAPc <- subset_samples(phy_obj_B_Soil, Soil == "Los_Angeles")
phy_BR_LAPc <- subset_samples(phy_BR_LAPc, Specie == "P_chilensis")

phy_BR_VPt <- subset_samples(phy_obj_B_Soil, Soil == "Vicuna")
phy_BR_VPt <- subset_samples(phy_BR_VPt, Specie == "P_tamarugo")

phy_BR_LSPt <- subset_samples(phy_obj_B_Soil, Soil == "La_Serena")
phy_BR_LSPt <- subset_samples(phy_BR_LSPt, Specie == "P_tamarugo")

phy_BR_LAPt <- subset_samples(phy_obj_B_Soil, Soil == "Los_Angeles")
phy_BR_LAPt <- subset_samples(phy_BR_LAPt, Specie == "P_tamarugo")
```

```{r}
phy_norm_B_Soil <- subset_samples(phy_norm, Sample == "Soil")

phy_norm_BS_mrg <- merge_samples(phy_norm_B_Soil, "Soil", fun = mean)

phy_norm_B_Root <- subset_samples(phy_norm, Sample == "Root")

sample_data(phy_norm_B_Root)[,"Soil_sp"] <- paste(sample_data(phy_norm_B_Root)$Soil, sample_data(phy_norm_B_Root)$Specie, sep = "_")

phy_norm_BR_mrg <- merge_samples(phy_norm_B_Root, "Soil_sp", fun = mean)
```

#VicuÃ±a
```{r}
common_V_VPc <- intersect(rownames(otu_table(phy_BS_V)), rownames(otu_table(phy_BR_VPc)))

common_V_VPt <- intersect(rownames(otu_table(phy_BS_V)), rownames(otu_table(phy_BR_VPt)))

Root_Pc_V <- t(otu_table(phy_norm_BR_mrg))[colnames(otu_table(phy_norm_BR_mrg)) %in% common_V_VPc]
Root_Pc_V <- as.data.frame(as.matrix(Root_Pc_V)) %>% dplyr::select(Vicuna_P_chilensis)

Soil_Pc_V <- t(otu_table(phy_norm_BS_mrg))[colnames(otu_table(phy_norm_BS_mrg)) %in% common_V_VPc]
Soil_Pc_V <- as.data.frame(as.matrix(Soil_Pc_V)) %>% dplyr::select(Vicuna)

Root_Pt_V <- t(otu_table(phy_norm_BR_mrg))[colnames(otu_table(phy_norm_BR_mrg)) %in% common_V_VPt]
Root_Pt_V <- as.data.frame(as.matrix(Root_Pt_V)) %>% dplyr::select(Vicuna_P_tamarugo)

Soil_Pt_V <- t(otu_table(phy_norm_BS_mrg))[colnames(otu_table(phy_norm_BS_mrg)) %in% common_V_VPt]
Soil_Pt_V <- as.data.frame(as.matrix(Soil_Pt_V)) %>% dplyr::select(Vicuna)

P_tam_v <- merge(Soil_Pt_V, Root_Pt_V, by= "row.names")
rownames(P_tam_v) <- P_tam_v$Row.names
P_tam_v[,"ASVs"] <- P_tam_v$Row.names
P_tam_v$ASVs <- as.factor(P_tam_v$ASVs)
P_tam_v[,"Delta"] <- P_tam_v$Vicuna_P_tamarugo-P_tam_v$Vicuna
P_tam_v <- P_tam_v[,-1]

P_tam_v <- merge(P_tam_v,tax_table(phy_norm_BR_mrg), by = "row.names")

ggplot(P_tam_v, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. tamarugo recruitment VicuÃ±a") +
  scale_y_discrete(labels=paste(P_tam_v$Kingdom,P_tam_v$Phylum,P_tam_v$Class,P_tam_v$Order,P_tam_v$Family,P_tam_v$Genus,P_tam_v$Specie,P_tam_v$ASVs))

length(P_tam_v$Delta[P_tam_v$Delta==0])
length(P_tam_v$Delta[P_tam_v$Delta>0])
length(P_tam_v$Delta[P_tam_v$Delta<0])

P_chil_v <- merge(Soil_Pc_V, Root_Pc_V, by= "row.names")
rownames(P_chil_v) <- P_chil_v$Row.names
P_chil_v[,"ASVs"] <- P_chil_v$Row.names
P_chil_v$ASVs <- as.factor(P_chil_v$ASVs)
P_chil_v[,"Delta"] <- P_chil_v$Vicuna_P_chilensis-P_chil_v$Vicuna
P_chil_v <- P_chil_v[,-1]

P_chil_v <- merge(P_chil_v,tax_table(phy_norm_BR_mrg), by = "row.names")

ggplot(P_chil_v, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. chilensis recruitment VicuÃ±a") +
  scale_y_discrete(labels=paste(P_chil_v$Kingdom,P_chil_v$Phylum,P_chil_v$Class,P_chil_v$Order,P_chil_v$Family,P_chil_v$Genus,P_chil_v$Specie,P_chil_v$ASVs))

length(P_chil_v$Delta[P_chil_v$Delta==0])
length(P_chil_v$Delta[P_chil_v$Delta>0])
length(P_chil_v$Delta[P_chil_v$Delta<0])

df1 <- P_tam_v[order(abs(P_tam_v$Delta),decreasing = TRUE),] %>% head(30)
df2 <- P_chil_v[order(abs(P_chil_v$Delta),decreasing = TRUE),] %>% head(30)

df3 <- union(df1$Row.names,df2$Row.names)
df1 <- P_tam_v[P_tam_v$Row.names %in% df3,]
df2 <- P_chil_v[P_chil_v$Row.names %in% df3,]

p1 <- ggplot(df2, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. chilensis recruitment VicuÃ±a") +
  scale_y_discrete(labels=paste(df2$Kingdom,df2$Phylum,df2$Class,df2$Order,df2$Family,df2$Genus,df2$Specie,df2$ASVs))

p2 <- ggplot(df1, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. tamarugo recruitment VicuÃ±a") +
  scale_y_discrete(labels=paste(df1$Kingdom,df1$Phylum,df1$Class,df1$Order,df1$Family,df1$Genus,df1$Specie,df1$ASVs))

grid.arrange(p1, p2, ncol=2)
```

## venn diagram
```{r}
#higher than zero
P_tam_V_hz <- P_tam_v %>% filter(Delta > 0)
P_tam_V_hz <- P_tam_V_hz$Row.names

P_chil_v_hz <- P_chil_v %>% filter(Delta > 0)
P_chil_v_hz <- P_chil_v_hz$Row.names

plot.new()
suppressWarnings(grid.draw(venn.diagram(
  list(P_tam_V_hz=P_tam_V_hz,P_chil_v_hz=P_chil_v_hz),
  filename=NULL, col=c("yellowgreen","green"), margin=c(0.35,0.35,0.35,0.35),
  category.names=c("P.tamarugo","P.chilensis"))))

#equal to zero
P_tam_V_ez <- P_tam_v %>% filter(Delta == 0)
P_tam_V_ez <- P_tam_V_ez$Row.names

P_chil_v_ez <- P_chil_v %>% filter(Delta == 0)
P_chil_v_ez <- P_chil_v_ez$Row.names

#lower than zero
P_tam_V_lz <- P_tam_v %>% filter(Delta < 0)
P_tam_V_lz <- P_tam_V_lz$Row.names

P_chil_v_lz <- P_chil_v %>% filter(Delta < 0)
P_chil_v_lz <- P_chil_v_lz$Row.names
```

###common recruitment
```{r}
common_V <- intersect(P_tam_V_hz, P_chil_v_hz)

common_V <- t(otu_table(phy_norm_BR_mrg))[colnames(otu_table(phy_norm_BR_mrg)) %in% common_V]

common_V <- as.data.frame(as.matrix(common_V))

common_V[,"Delta"] <- common_V$Vicuna_P_chilensis-common_V$Vicuna_P_tamarugo

common_V <- merge(common_V,tax_table(phy_norm_BR_mrg), by = "row.names")
rownames(common_V) <- common_V$Row.names
common_V <- common_V[,-1]
common_V[,"ASVs"] <- rownames(common_V)

ggplot(common_V, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-1, 1), low = "yellowgreen", mid = "gray", high = "green") +
  geom_point() + geom_point(size = 5.5) + xlim(-1, 1) +
  ggtitle("Common recruited VicuÃ±a") +
  scale_y_discrete(labels=paste(common_V$Kingdom,common_V$Phylum,common_V$Class,common_V$Order,common_V$Family,common_V$Genus,common_V$Specie,common_V$ASVs))

common_V2 <- common_V[order(abs(common_V$Delta),decreasing = TRUE),] %>% head(30)

ggplot(common_V2, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-1, 1), low = "yellowgreen", mid = "gray", high = "green") +
  geom_point() + geom_point(size = 5.5) + xlim(-1, 1) +
  ggtitle("Common recruited VicuÃ±a") +
  scale_y_discrete(labels=paste(common_V2$Kingdom,common_V2$Phylum,common_V2$Class,common_V2$Order,common_V2$Family,common_V2$Genus,common_V2$Specie,common_V2$ASVs))
```

#La Serena
```{r}
common_LS_LSPc <- intersect(rownames(otu_table(phy_BS_LS)), rownames(otu_table(phy_BR_LSPc)))

common_LS_LSPt <- intersect(rownames(otu_table(phy_BS_LS)), rownames(otu_table(phy_BR_LSPt)))

Root_Pc_LS <- t(otu_table(phy_norm_BR_mrg))[colnames(otu_table(phy_norm_BR_mrg)) %in% common_LS_LSPc]
Root_Pc_LS <- as.data.frame(as.matrix(Root_Pc_LS)) %>% dplyr::select(La_Serena_P_chilensis)

Soil_Pc_LS <- t(otu_table(phy_norm_BS_mrg))[colnames(otu_table(phy_norm_BS_mrg)) %in% common_LS_LSPc]
Soil_Pc_LS <- as.data.frame(as.matrix(Soil_Pc_LS)) %>% dplyr::select(La_Serena)

Root_Pt_LS <- t(otu_table(phy_norm_BR_mrg))[colnames(otu_table(phy_norm_BR_mrg)) %in% common_LS_LSPt]
Root_Pt_LS <- as.data.frame(as.matrix(Root_Pt_LS)) %>% dplyr::select(La_Serena_P_tamarugo)

Soil_Pt_LS <- t(otu_table(phy_norm_BS_mrg))[colnames(otu_table(phy_norm_BS_mrg)) %in% common_LS_LSPt]
Soil_Pt_LS <- as.data.frame(as.matrix(Soil_Pt_LS)) %>% dplyr::select(La_Serena)

P_tam_LS <- merge(Soil_Pt_LS, Root_Pt_LS, by= "row.names")
rownames(P_tam_LS) <- P_tam_LS$Row.names
P_tam_LS[,"ASVs"] <- P_tam_LS$Row.names
P_tam_LS$ASVs <- as.factor(P_tam_LS$ASVs)
P_tam_LS[,"Delta"] <- P_tam_LS$La_Serena_P_tamarugo-P_tam_LS$La_Serena
P_tam_LS <- P_tam_LS[,-1]

P_tam_LS <- merge(P_tam_LS,tax_table(phy_norm_BR_mrg), by = "row.names")

ggplot(P_tam_LS, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. tamarugo recruitment La Serena") +
  scale_y_discrete(labels=paste(P_tam_LS$Kingdom,P_tam_LS$Phylum,P_tam_LS$Class,P_tam_LS$Order,P_tam_LS$Family,P_tam_LS$Genus,P_tam_LS$Specie,P_tam_LS$ASVs))

length(P_tam_LS$Delta[P_tam_LS$Delta==0])
length(P_tam_LS$Delta[P_tam_LS$Delta>0])
length(P_tam_LS$Delta[P_tam_LS$Delta<0])

P_chil_LS <- merge(Soil_Pc_LS, Root_Pc_LS, by= "row.names")
rownames(P_chil_LS) <- P_chil_LS$Row.names
P_chil_LS[,"ASVs"] <- P_chil_LS$Row.names
P_chil_LS$ASVs <- as.factor(P_chil_LS$ASVs)
P_chil_LS[,"Delta"] <- P_chil_LS$La_Serena_P_chilensis-P_chil_LS$La_Serena
P_chil_LS <- P_chil_LS[,-1]

P_chil_LS <- merge(P_chil_LS,tax_table(phy_norm_BR_mrg), by = "row.names")

ggplot(P_chil_LS, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. chilensis recruitment La Serena") +
  scale_y_discrete(labels=paste(P_chil_LS$Kingdom,P_chil_LS$Phylum,P_chil_LS$Class,P_chil_LS$Order,P_chil_LS$Family,P_chil_LS$Genus,P_chil_LS$Specie,P_chil_LS$ASVs))

length(P_chil_LS$Delta[P_chil_LS$Delta==0])
length(P_chil_LS$Delta[P_chil_LS$Delta>0])
length(P_chil_LS$Delta[P_chil_LS$Delta<0])

df1 <- P_tam_LS[order(abs(P_tam_LS$Delta),decreasing = TRUE),] %>% head(30)
df2 <- P_chil_LS[order(abs(P_chil_LS$Delta),decreasing = TRUE),] %>% head(30)

df3 <- union(df1$Row.names,df2$Row.names)
df1 <- P_tam_LS[P_tam_LS$Row.names %in% df3,]
df2 <- P_chil_LS[P_chil_LS$Row.names %in% df3,]

p1 <- ggplot(df2, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. chilensis recruitment La Serena") +
  scale_y_discrete(labels=paste(df2$Kingdom,df2$Phylum,df2$Class,df2$Order,df2$Family,df2$Genus,df2$Specie,df2$ASVs))

p2 <- ggplot(df1, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. tamarugo recruitment La Serena") +
  scale_y_discrete(labels=paste(df1$Kingdom,df1$Phylum,df1$Class,df1$Order,df1$Family,df1$Genus,df1$Specie,df1$ASVs))

grid.arrange(p1, p2, ncol=2)
```

##venn diagram
```{r}
#higher than zero
P_tam_LS_hz <- P_tam_LS %>% filter(Delta > 0)
P_tam_LS_hz <- P_tam_LS_hz$Row.names

P_chil_LS_hz <- P_chil_LS %>% filter(Delta > 0)
P_chil_LS_hz <- P_chil_LS_hz$Row.names

plot.new()
suppressWarnings(grid.draw(venn.diagram(
  list(P_tam_LS_hz=P_tam_LS_hz,P_chil_LS_hz=P_chil_LS_hz),
  filename=NULL, col=c("yellowgreen","green"), margin=c(0.35,0.35,0.35,0.35),
  category.names=c("P.tamarugo","P.chilensis"))))

#equal to zero
P_tam_LS_ez <- P_tam_LS %>% filter(Delta == 0)
P_tam_LS_ez <- P_tam_LS_ez$Row.names

P_chil_LS_ez <- P_chil_LS %>% filter(Delta == 0)
P_chil_LS_ez <- P_chil_LS_ez$Row.names

#equal to zero
P_tam_LS_lz <- P_tam_LS %>% filter(Delta < 0)
P_tam_LS_lz <- P_tam_LS_lz$Row.names

P_chil_LS_lz <- P_chil_LS %>% filter(Delta < 0)
P_chil_LS_lz <- P_chil_LS_lz$Row.names
```

###common recruitment
```{r}
common_LS <- intersect(P_tam_LS_hz, P_chil_LS_hz)

common_LS <- t(otu_table(phy_norm_BR_mrg))[colnames(otu_table(phy_norm_BR_mrg)) %in% common_LS]

common_LS <- as.data.frame(as.matrix(common_LS))

common_LS[,"Delta"] <- common_LS$La_Serena_P_chilensis-common_LS$La_Serena_P_tamarugo

common_LS <- merge(common_LS,tax_table(phy_norm_BR_mrg), by = "row.names")
rownames(common_LS) <- common_LS$Row.names
common_LS <- common_LS[,-1]
common_LS[,"ASVs"] <- rownames(common_LS)

ggplot(common_LS, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-1, 1), low = "yellowgreen", mid = "gray", high = "green") +
  geom_point() + geom_point(size = 5.5) + xlim(-1, 1) +
  ggtitle("Common recruited VicuÃ±a") +
  scale_y_discrete(labels=paste(common_LS$Kingdom,common_LS$Phylum,common_LS$Class,common_LS$Order,common_LS$Family,common_LS$Genus,common_LS$Specie,common_LS$ASVs))

common_LS2 <- common_LS[order(abs(common_LS$Delta),decreasing = TRUE),] %>% head(30)

ggplot(common_LS2, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-1, 1), low = "yellowgreen", mid = "gray", high = "green") +
  geom_point() + geom_point(size = 5.5) + xlim(-1, 1) +
  ggtitle("Common recruited VicuÃ±a") +
  scale_y_discrete(labels=paste(common_LS2$Kingdom,common_LS2$Phylum,common_LS2$Class,common_LS2$Order,common_LS2$Family,common_LS2$Genus,common_LS2$Specie,common_LS2$ASVs))
```

#Los Angeles
```{r}
common_LA_LAPc <- intersect(rownames(otu_table(phy_BS_LA)), rownames(otu_table(phy_BR_LAPc)))

common_LA_LAPt <- intersect(rownames(otu_table(phy_BS_LA)), rownames(otu_table(phy_BR_LAPt)))

Root_Pc_LA <- t(otu_table(phy_norm_BR_mrg))[colnames(otu_table(phy_norm_BR_mrg)) %in% common_LA_LAPc]
Root_Pc_LA <- as.data.frame(as.matrix(Root_Pc_LA)) %>% dplyr::select(Los_Angeles_P_chilensis)

Soil_Pc_LA <- t(otu_table(phy_norm_BS_mrg))[colnames(otu_table(phy_norm_BS_mrg)) %in% common_LA_LAPc]
Soil_Pc_LA <- as.data.frame(as.matrix(Soil_Pc_LA)) %>% dplyr::select(Los_Angeles)

Root_Pt_LA <- t(otu_table(phy_norm_BR_mrg))[colnames(otu_table(phy_norm_BR_mrg)) %in% common_LA_LAPt]
Root_Pt_LA <- as.data.frame(as.matrix(Root_Pt_LA)) %>% dplyr::select(Los_Angeles_P_tamarugo)

Soil_Pt_LA <- t(otu_table(phy_norm_BS_mrg))[colnames(otu_table(phy_norm_BS_mrg)) %in% common_LA_LAPt]
Soil_Pt_LA <- as.data.frame(as.matrix(Soil_Pt_LA)) %>% dplyr::select(Los_Angeles)

P_tam_LA <- merge(Soil_Pt_LA, Root_Pt_LA, by= "row.names")
rownames(P_tam_LA) <- P_tam_LA$Row.names
P_tam_LA[,"ASVs"] <- P_tam_LA$Row.names
P_tam_LA$ASVs <- as.factor(P_tam_LA$ASVs)
P_tam_LA[,"Delta"] <- P_tam_LA$Los_Angeles_P_tamarugo-P_tam_LA$Los_Angeles
P_tam_LA <- P_tam_LA[,-1]

P_tam_LA <- merge(P_tam_LA,tax_table(phy_norm_BR_mrg), by = "row.names")

ggplot(P_tam_LA, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. tamarugo recruitment Los Angeles") +
  scale_y_discrete(labels=paste(P_tam_LA$Kingdom,P_tam_LA$Phylum,P_tam_LA$Class,P_tam_LA$Order,P_tam_LA$Family,P_tam_LA$Genus,P_tam_LA$Specie,P_tam_LA$ASVs))

length(P_tam_LA$Delta[P_tam_LA$Delta==0])
length(P_tam_LA$Delta[P_tam_LA$Delta>0])
length(P_tam_LA$Delta[P_tam_LA$Delta<0])

P_chil_LA <- merge(Soil_Pc_LA, Root_Pc_LA, by= "row.names")
rownames(P_chil_LA) <- P_chil_LA$Row.names
P_chil_LA[,"ASVs"] <- P_chil_LA$Row.names
P_chil_LA$ASVs <- as.factor(P_chil_LA$ASVs)
P_chil_LA[,"Delta"] <- P_chil_LA$Los_Angeles_P_chilensis-P_chil_LA$Los_Angeles
P_chil_LA <- P_chil_LA[,-1]

P_chil_LA <- merge(P_chil_LA,tax_table(phy_norm_BR_mrg), by = "row.names")

ggplot(P_chil_LA, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. chilensis recruitment Los Angeles") +
  scale_y_discrete(labels=paste(P_chil_LA$Kingdom,P_chil_LA$Phylum,P_chil_LA$Class,P_chil_LA$Order,P_chil_LA$Family,P_chil_LA$Genus,P_chil_LA$Specie,P_chil_LA$ASVs))

length(P_chil_LA$Delta[P_chil_LA$Delta==0])
length(P_chil_LA$Delta[P_chil_LA$Delta>0])
length(P_chil_LA$Delta[P_chil_LA$Delta<0])

df1 <- P_tam_LA[order(abs(P_tam_LA$Delta),decreasing = TRUE),] %>% head(30)
df2 <- P_chil_LA[order(abs(P_chil_LA$Delta),decreasing = TRUE),] %>% head(30)

df3 <- union(df1$Row.names,df2$Row.names)
df1 <- P_tam_LA[P_tam_LA$Row.names %in% df3,]
df2 <- P_chil_LA[P_chil_LA$Row.names %in% df3,]

p1 <- ggplot(df2, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. chilensis recruitment Los Angeles") +
  scale_y_discrete(labels=paste(df2$Kingdom,df2$Phylum,df2$Class,df2$Order,df2$Family,df2$Genus,df2$Specie,df2$ASVs))

p2 <- ggplot(df1, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-5, 3), low = "blue", mid = "gray", high = "red") +
  geom_point() + geom_point(size = 5.5) + xlim(-5, 3) +
  ggtitle("P. tamarugo recruitment Los Angeles") +
  scale_y_discrete(labels=paste(df1$Kingdom,df1$Phylum,df1$Class,df1$Order,df1$Family,df1$Genus,df1$Specie,df1$ASVs))

grid.arrange(p1, p2, ncol=2)
```

##venn diagram
```{r}
#higher than zero
P_tam_LA_hz <- P_tam_LA %>% filter(Delta > 0)
P_tam_LA_hz <- P_tam_LA_hz$Row.names

P_chil_LA_hz <- P_chil_LA %>% filter(Delta > 0)
P_chil_LA_hz <- P_chil_LA_hz$Row.names

#pdf("Manuscript/Plots/Supplementary/Venn_LA_Bacteria_hz.pdf", width = 10, height = 10)
plot.new()
suppressWarnings(grid.draw(venn.diagram(
  list(P_tam_LA_hz=P_tam_LA_hz,P_chil_LA_hz=P_chil_LA_hz),
  filename=NULL, col=c("yellowgreen","green"), margin=c(0.35,0.35,0.35,0.35),
  category.names=c("P.tamarugo","P.chilensis"))))
#dev.off()

#equal to zero
P_tam_LA_ez <- P_tam_LA %>% filter(Delta == 0)
P_tam_LA_ez <- P_tam_LA_ez$Row.names

P_chil_LA_ez <- P_chil_LA %>% filter(Delta == 0)
P_chil_LA_ez <- P_chil_LA_ez$Row.names

#lower than zero
P_tam_LA_lz <- P_tam_LA %>% filter(Delta < 0)
P_tam_LA_lz <- P_tam_LA_lz$Row.names

P_chil_LA_lz <- P_chil_LA %>% filter(Delta < 0)
P_chil_LA_lz <- P_chil_LA_lz$Row.names
```

###common recruitment
```{r}
common_LA <- intersect(P_tam_LA_hz, P_chil_LA_hz)

common_LA <- t(otu_table(phy_norm_BR_mrg))[colnames(otu_table(phy_norm_BR_mrg)) %in% common_LA]

common_LA <- as.data.frame(as.matrix(common_LA))

common_LA[,"Delta"] <- common_LA$Los_Angeles_P_chilensis-common_LA$Los_Angeles_P_tamarugo

common_LA <- merge(common_LA,tax_table(phy_norm_BR_mrg), by = "row.names")
rownames(common_LA) <- common_LA$Row.names
common_LA <- common_LA[,-1]
common_LA[,"ASVs"] <- rownames(common_LA)

ggplot(common_LA, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  theme(axis.text = element_blank()) +
  scale_colour_gradient2(midpoint = 0, limits = c(-1, 1), low = "yellowgreen", mid = "gray", high = "green") +
  geom_point() + geom_point(size = 5.5) + xlim(-1, 1) +
  ggtitle("Common recruited VicuÃ±a") +
  scale_y_discrete(labels=paste(common_LA$Kingdom,common_LA$Phylum,common_LA$Class,common_LA$Order,common_LA$Family,common_LA$Genus,common_LA$Specie,common_LA$ASVs))

common_LA2 <- common_LA[order(abs(common_LA$Delta),decreasing = TRUE),] %>% head(30)

ggplot(common_LA2, aes(Delta,ASVs, color = Delta)) +
  theme_bw() + 
  scale_colour_gradient2(midpoint = 0, limits = c(-1, 1), low = "yellowgreen", mid = "gray", high = "green") +
  geom_point() + geom_point(size = 5.5) + xlim(-1, 1) +
  ggtitle("Common recruited Los angeles") +
  scale_y_discrete(labels=paste(common_LA2$Kingdom,common_LA2$Phylum,common_LA2$Class,common_LA2$Order,common_LA2$Family,common_LA2$Genus,common_LA2$Specie,common_LA2$ASVs))
```

#Session info
```{r}
sessionInfo()
```